<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Rodrigo Amadeu" />

<meta name="date" content="2023-10-03" />

<title>AGHmatrix Tutorial</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">AGHmatrix Tutorial</h1>
<h4 class="author">Rodrigo Amadeu</h4>
<h4 class="date">2023-10-03</h4>



<div id="contact" class="section level2">
<h2>Contact</h2>
<p>Rodrigo R Amadeu<br />
rramadeu at gmail dot com<br />
<a href="https://rramadeu.github.io" class="uri">https://rramadeu.github.io</a></p>
</div>
<div id="overview" class="section level2">
<h2>Overview</h2>
<p>AGHmatrix software is an R-package to build relationship matrices
using pedigree (A matrix) and/or molecular markers (G matrix) with the
possibility to build a combined matrix of Pedigree corrected by
Molecular (H matrix). The package works with diploid and autopolyploid
data.</p>
</div>
<div id="matrices-computation-implemented-in-the-aghmatrix" class="section level2">
<h2>Matrices computation implemented in the <code>AGHmatrix</code></h2>
<p>Currently the package computes the following 17 different
relationship matrices:</p>
<div id="pedigree-based-relationship-matrix-a-matrix" class="section level3">
<h3>Pedigree-based relationship matrix (A matrix)</h3>
<table>
<thead>
<tr class="header">
<th></th>
<th>Additive</th>
<th>Non-Additive</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><strong>Diploid</strong></td>
<td>Henderson (1976)</td>
<td>Cockerham (1954)</td>
</tr>
<tr class="even">
<td><strong>Autopolyploid</strong></td>
<td>Kerr (2012), Slater (2013)</td>
<td></td>
</tr>
</tbody>
</table>
</div>
<div id="molecular-based-relationship-matrix-g-matrix" class="section level3">
<h3>Molecular-based relationship matrix (G matrix)</h3>
<table>
<colgroup>
<col width="20%" />
<col width="45%" />
<col width="34%" />
</colgroup>
<thead>
<tr class="header">
<th></th>
<th>Additive</th>
<th>Non-Additive</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><strong>Diploid</strong></td>
<td>Yang (2010), VanRaden (2012), Liu (2020)</td>
<td>Su (2012), Vitezica (2013)</td>
</tr>
<tr class="even">
<td><strong>Polyploid</strong></td>
<td>Slater (2016), de Bem Oliveira (2019)</td>
<td>Slater (2016), Endelman (2018)</td>
</tr>
</tbody>
</table>
</div>
<div id="combined-pedigree-and-molecular-based-relationship-matrix-h-matrix" class="section level3">
<h3>Combined pedigree and molecular-based relationship matrix (H
matrix)</h3>
<table>
<thead>
<tr class="header">
<th><strong>Any ploidy/effect</strong></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Legarra (2009), Munoz (2014), Martini (2018)</td>
</tr>
</tbody>
</table>
<p>Additionally there is a beta implementation to compute A matrix when
parentage is not deterministic as in a polycross design. See
<code>?AmatrixPolycross</code>.</p>
</div>
</div>
<div id="citation" class="section level2">
<h2>Citation</h2>
<p>To cite this R package:</p>
<p>Amadeu RR, Garcia AA, Munoz PR, Ferrão LF. AGHmatrix: genetic
relationship matrices in R. Bioinformatics. 2023 Jul 1;39(7):btad445. <a href="https://doi.org/10.1093/bioinformatics/btad445" class="uri">https://doi.org/10.1093/bioinformatics/btad445</a></p>
</div>
<div id="installing-and-loading" class="section level2">
<h2>Installing and loading</h2>
<p>Within R:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="do">## Install stable version</span></span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;AGHmatrix&quot;</span>)</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a></span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a><span class="do">## Install development version</span></span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;devtools&quot;</span>)</span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">&quot;rramadeu/AGHmatrix&quot;</span>)</span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a></span>
<span id="cb1-8"><a href="#cb1-8" tabindex="-1"></a><span class="do">## Load </span></span>
<span id="cb1-9"><a href="#cb1-9" tabindex="-1"></a><span class="fu">library</span>(AGHmatrix)</span></code></pre></div>
</div>
<div id="relationship-matrices-using-pedigree-data---a-matrix" class="section level2">
<h2>Relationship matrices using pedigree data - A matrix</h2>
<p><code>Amatrix</code> process the pedigree and build the A-matrix
related to that given pedigree. The matrix is built based in the
recursive method presented in Mrode (2014) and described by Henderson
(1976). This method is expanded for higher ploidies (n-ploidy) as
detailed in Kerr et al. (2012). After loading the package you have to
load your data file into the software. To do this, you can use the
function <code>read.data()</code> or <code>read.csv()</code> for
example. Your data should be available in R as a <code>data.frame</code>
structure in the following order: column 1 must be the
individual/genotype names (id), columns 2 and 3 must be the parent
names. For the algorithm, it does not matter who is the mother and who
is the father (so, no sex column). There is a pedigree data example
(<code>ped.mrode</code>) that can be used to look at the structure and
order the data.</p>
<p>To load <code>ped.mrode</code>:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="fu">data</span>(ped.mrode)</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a>ped.mrode</span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a><span class="co">#&gt;    Ind Par1 Par2</span></span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a><span class="co">#&gt; 1 Anc1    0    0</span></span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a><span class="co">#&gt; 2 Anc2    0    0</span></span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a><span class="co">#&gt; 3 Var1 Anc1 Anc2</span></span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a><span class="co">#&gt; 4 Var2 Anc1    0</span></span>
<span id="cb2-8"><a href="#cb2-8" tabindex="-1"></a><span class="co">#&gt; 5 Var3 Var2 Var1</span></span>
<span id="cb2-9"><a href="#cb2-9" tabindex="-1"></a><span class="co">#&gt; 6 Var4 Var3 Anc2</span></span>
<span id="cb2-10"><a href="#cb2-10" tabindex="-1"></a><span class="fu">str</span>(ped.mrode) <span class="co">#check the structure</span></span>
<span id="cb2-11"><a href="#cb2-11" tabindex="-1"></a><span class="co">#&gt; &#39;data.frame&#39;:    6 obs. of  3 variables:</span></span>
<span id="cb2-12"><a href="#cb2-12" tabindex="-1"></a><span class="co">#&gt;  $ Ind : Factor w/ 6 levels &quot;Anc1&quot;,&quot;Anc2&quot;,..: 1 2 3 4 5 6</span></span>
<span id="cb2-13"><a href="#cb2-13" tabindex="-1"></a><span class="co">#&gt;  $ Par1: Factor w/ 4 levels &quot;0&quot;,&quot;Anc1&quot;,&quot;Var2&quot;,..: 1 1 2 2 3 4</span></span>
<span id="cb2-14"><a href="#cb2-14" tabindex="-1"></a><span class="co">#&gt;  $ Par2: Factor w/ 3 levels &quot;0&quot;,&quot;Anc2&quot;,&quot;Var1&quot;: 1 1 2 1 3 2</span></span></code></pre></div>
<p>The example <code>ped.mrode</code> has 3 columns, column 1 contains
the names of the individual/genotypes, column 2 contains the names of
the first parent, column 3 contains the names of the second parental
(example from Table 2.1 of Mrode 2014). There is no header template, and
the unknown value must be equal 0. Your data has to be in the same
format of <code>ped.mrode</code>.</p>
<p>Internally the algorithm first pre-process the pedigree: the
individuals are numerated <span class="math inline">\(1\)</span> to
<span class="math inline">\(n\)</span>. Then, it is verified whether the
genotypes in the pedigree are in chronological order (i.e. if the
parents of a given individual are located before to this individual in
the pedigree data set). If this order is not followed, the algorithm
performs the necessary changes to correct them in a iterative way. After
this pre-processing, the matrix computation proceeds as in Henderson
(1976) for diploid - for additive or dominance relationship - and as in
Kerr et al. (2012) for autotetraploids - for additive relationship. For
autotetraploids, there is the option to include double-reduction
fraction. For diploids there is the option to compute the dominant
relationship matrix (Cockerham, 1954).</p>
<p>It follows some usage examples with the <code>ped.mrode</code>.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="co">#Computing additive relationship matrix for diploids (Henderson 1976):</span></span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a><span class="fu">Amatrix</span>(ped.mrode, <span class="at">ploidy=</span><span class="dv">2</span>)</span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a></span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a><span class="co">#Computing dominant relationship matrix for diploids (Cockerham 1954):</span></span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a><span class="fu">Amatrix</span>(ped.mrode, <span class="at">ploidy=</span><span class="dv">2</span>, <span class="at">dominance=</span><span class="cn">TRUE</span>)</span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a></span>
<span id="cb3-7"><a href="#cb3-7" tabindex="-1"></a><span class="co">#Computing additive relationship matrix for autotetraploids (Kerr 2012):</span></span>
<span id="cb3-8"><a href="#cb3-8" tabindex="-1"></a><span class="fu">Amatrix</span>(ped.mrode, <span class="at">ploidy=</span><span class="dv">4</span>)</span>
<span id="cb3-9"><a href="#cb3-9" tabindex="-1"></a></span>
<span id="cb3-10"><a href="#cb3-10" tabindex="-1"></a><span class="co">#Computing additive relationship matrix for autooctaploids (Kerr 2012):</span></span>
<span id="cb3-11"><a href="#cb3-11" tabindex="-1"></a><span class="fu">Amatrix</span>(ped.mrode, <span class="at">ploidy=</span><span class="dv">8</span>)</span>
<span id="cb3-12"><a href="#cb3-12" tabindex="-1"></a></span>
<span id="cb3-13"><a href="#cb3-13" tabindex="-1"></a><span class="co">#Computing additive relationship matrix for autotetraploids</span></span>
<span id="cb3-14"><a href="#cb3-14" tabindex="-1"></a><span class="co"># and double-reduction of 0.1 (Kerr 2012):</span></span>
<span id="cb3-15"><a href="#cb3-15" tabindex="-1"></a><span class="fu">Amatrix</span>(ped.mrode, <span class="at">ploidy=</span><span class="dv">4</span>, <span class="at">w=</span><span class="fl">0.1</span>)</span>
<span id="cb3-16"><a href="#cb3-16" tabindex="-1"></a></span>
<span id="cb3-17"><a href="#cb3-17" tabindex="-1"></a><span class="co">#Computing additive relationship matrix for autotetraploids</span></span>
<span id="cb3-18"><a href="#cb3-18" tabindex="-1"></a><span class="co"># and double-reduction of 0.1 as in Slater et al. (2014):</span></span>
<span id="cb3-19"><a href="#cb3-19" tabindex="-1"></a><span class="fu">Amatrix</span>(ped.mrode, <span class="at">ploidy=</span><span class="dv">4</span>, <span class="at">w=</span><span class="fl">0.1</span>, <span class="at">slater =</span> <span class="cn">TRUE</span>) </span>
<span id="cb3-20"><a href="#cb3-20" tabindex="-1"></a><span class="co">#not recommended, but kept in the package to reproduce some former analysis</span></span>
<span id="cb3-21"><a href="#cb3-21" tabindex="-1"></a></span>
<span id="cb3-22"><a href="#cb3-22" tabindex="-1"></a><span class="co">#Computing additive relationship matrix for autohexaploids</span></span>
<span id="cb3-23"><a href="#cb3-23" tabindex="-1"></a><span class="co"># and double-reduction of 0.1 (Kerr 2012):</span></span>
<span id="cb3-24"><a href="#cb3-24" tabindex="-1"></a><span class="fu">Amatrix</span>(ped.mrode, <span class="at">ploidy=</span><span class="dv">6</span>, <span class="at">w=</span><span class="fl">0.1</span>)</span></code></pre></div>
<p>More information about <code>Amatrix</code> can be found with:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a>?Amatrix</span></code></pre></div>
</div>
<div id="diploid-g-matrix-relationship-matrices-using-the-molecular-data" class="section level2">
<h2>Diploid G matrix: relationship matrices using the molecular
data</h2>
<p><code>Gmatrix</code> handles the molecular-marker matrix and builds
the relationship matrix. Molecular markers data should be organized in a
matrix format (individuals in rows and markers in columns) coded as 0,
1, 2 and missing data value (numeric or <code>NA</code>). Import your
molecular marker data into <code>R</code> with the function
<code>read.table()</code> or <code>read.csv()</code> and convert to a
matrix format with the function <code>as.matrix()</code>. The function
<code>Gmatrix</code> can be used to construct the additive relationship
either as proposed by Yang et al. (2010) or the proposed by VanRaden
(2008). The function can also construct the dominance relationship
matrix either as proposed by Su et al. (2012) or as proposed by Vitezica
et al. (2013). As an example, here we build the four matrices using real
data from Resende et al. (2012).</p>
<p>To load <code>snp.pine</code> and to check its structure:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="fu">data</span>(snp.pine)</span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a>snp.pine[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a><span class="co">#&gt;         0-10024-01-114 0-10037-01-257 0-10040-02-394 0-10040-02-41</span></span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a><span class="co">#&gt; 1087120             -9              1             -9            -9</span></span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a><span class="co">#&gt; 1085618              2              2              2             2</span></span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a><span class="co">#&gt; 1091040              2              1              2             2</span></span>
<span id="cb5-7"><a href="#cb5-7" tabindex="-1"></a><span class="co">#&gt; 1091686              2              1              2             2</span></span>
<span id="cb5-8"><a href="#cb5-8" tabindex="-1"></a><span class="co">#&gt; 1082624              2              2              0             1</span></span>
<span id="cb5-9"><a href="#cb5-9" tabindex="-1"></a><span class="co">#&gt;         0-10044-01-392</span></span>
<span id="cb5-10"><a href="#cb5-10" tabindex="-1"></a><span class="co">#&gt; 1087120             -9</span></span>
<span id="cb5-11"><a href="#cb5-11" tabindex="-1"></a><span class="co">#&gt; 1085618              2</span></span>
<span id="cb5-12"><a href="#cb5-12" tabindex="-1"></a><span class="co">#&gt; 1091040              2</span></span>
<span id="cb5-13"><a href="#cb5-13" tabindex="-1"></a><span class="co">#&gt; 1091686              2</span></span>
<span id="cb5-14"><a href="#cb5-14" tabindex="-1"></a><span class="co">#&gt; 1082624              1</span></span>
<span id="cb5-15"><a href="#cb5-15" tabindex="-1"></a><span class="fu">str</span>(snp.pine)</span>
<span id="cb5-16"><a href="#cb5-16" tabindex="-1"></a><span class="co">#&gt;  int [1:926, 1:4853] -9 2 2 2 2 2 2 2 2 2 ...</span></span>
<span id="cb5-17"><a href="#cb5-17" tabindex="-1"></a><span class="co">#&gt;  - attr(*, &quot;dimnames&quot;)=List of 2</span></span>
<span id="cb5-18"><a href="#cb5-18" tabindex="-1"></a><span class="co">#&gt;   ..$ : chr [1:926] &quot;1087120&quot; &quot;1085618&quot; &quot;1091040&quot; &quot;1091686&quot; ...</span></span>
<span id="cb5-19"><a href="#cb5-19" tabindex="-1"></a><span class="co">#&gt;   ..$ : chr [1:4853] &quot;0-10024-01-114&quot; &quot;0-10037-01-257&quot; &quot;0-10040-02-394&quot; &quot;0-10040-02-41&quot; ...</span></span></code></pre></div>
<p>In this dataset, we have 926 individuals with 4853 markers and the
missing data value is <code>-9</code>.</p>
<p>It follows some examples with the <code>snp.pine</code> data where
the unknown value (<code>missingValue</code>) is <code>-9</code>. Here
we set minimum allele frequency to <code>0.05</code>, so markers with
minor allele frequency lower than 0.05 are removed from the dataset
prior to the G matrix construction.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="co">#Computing the additive relationship matrix based on VanRaden 2008</span></span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a>G_VanRadenPine <span class="ot">&lt;-</span> <span class="fu">Gmatrix</span>(<span class="at">SNPmatrix=</span>snp.pine, <span class="at">missingValue=</span><span class="sc">-</span><span class="dv">9</span>, </span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a>                          <span class="at">maf=</span><span class="fl">0.05</span>, <span class="at">method=</span><span class="st">&quot;VanRaden&quot;</span>)</span>
<span id="cb6-4"><a href="#cb6-4" tabindex="-1"></a></span>
<span id="cb6-5"><a href="#cb6-5" tabindex="-1"></a><span class="co">#Computing the additive relationship matrix based on Yang 2010</span></span>
<span id="cb6-6"><a href="#cb6-6" tabindex="-1"></a>G_YangPine <span class="ot">&lt;-</span> <span class="fu">Gmatrix</span>(<span class="at">SNPmatrix=</span>snp.pine, <span class="at">missingValue=</span><span class="sc">-</span><span class="dv">9</span>, </span>
<span id="cb6-7"><a href="#cb6-7" tabindex="-1"></a>                      <span class="at">maf=</span><span class="fl">0.05</span>, <span class="at">method=</span><span class="st">&quot;Yang&quot;</span>)</span>
<span id="cb6-8"><a href="#cb6-8" tabindex="-1"></a></span>
<span id="cb6-9"><a href="#cb6-9" tabindex="-1"></a><span class="co">#Computing the dominance relationship matrix based on Su 2012</span></span>
<span id="cb6-10"><a href="#cb6-10" tabindex="-1"></a>G_SuPine <span class="ot">&lt;-</span> <span class="fu">Gmatrix</span>(<span class="at">SNPmatrix=</span>snp.pine, <span class="at">missingValue=</span><span class="sc">-</span><span class="dv">9</span>, </span>
<span id="cb6-11"><a href="#cb6-11" tabindex="-1"></a>                    <span class="at">maf=</span><span class="fl">0.05</span>, <span class="at">method=</span><span class="st">&quot;Su&quot;</span>)</span>
<span id="cb6-12"><a href="#cb6-12" tabindex="-1"></a></span>
<span id="cb6-13"><a href="#cb6-13" tabindex="-1"></a><span class="co">#Computing the dominance relationship matrix based on Vitezica 2013</span></span>
<span id="cb6-14"><a href="#cb6-14" tabindex="-1"></a>G_VitezicaPine <span class="ot">&lt;-</span> <span class="fu">Gmatrix</span>(<span class="at">SNPmatrix=</span>snp.pine, <span class="at">missingValue=</span><span class="sc">-</span><span class="dv">9</span>, </span>
<span id="cb6-15"><a href="#cb6-15" tabindex="-1"></a>                          <span class="at">maf=</span><span class="fl">0.05</span>, <span class="at">method=</span><span class="st">&quot;Vitezica&quot;</span>)</span></code></pre></div>
<p>More information about <code>Gmatrix</code> can be found with:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a>?Gmatrix</span></code></pre></div>
</div>
<div id="autopolyploid-g-matrix-relationship-matrices-using-the-molecular-data" class="section level2">
<h2>Autopolyploid G matrix: relationship matrices using the molecular
data</h2>
<p>Molecular markers data should be organized in a matrix format
(individual in rows and markers in columns) coded according to the
dosage level: 0, 1, 2, …, ploidy level, and missing data value (numeric
user-defined or <code>NA</code>). As an example, an autotetraploid
should be coded as 0, 1, 2, 3, 4, and a missing data value. In
autopolyploids, the function <code>Gmatrix</code> can be used to
construct: i) the additive relationship based on VanRaden (2008) and
extended by Ashraf (2016); ii) the full-autopolyploid including additive
and non-additive model as equations 8 and 9 in Slater et al. (2016);
iii) the pseudo-diploid model as equations 5, 6, and 7 Slater et
al. (2016). iv) the digenic-dominant model based on Endelman et
al. (2018). As an example, here we build the matrices using data from
Endelman et al. (2018) (<code>snp.sol</code>). There is also an option
to build weighted relationship matrices as in Liu et al. (2020).</p>
<p>The argument <code>ploidy.correction</code> defines the denominator
of the formula for the <code>VanRaden</code> method. If
<code>ploidy.correction=TRUE</code>, it uses the parametric correction
as <span class="math inline">\(\sum_i p f_i(1-f_i)\)</span>, where <span class="math inline">\(p\)</span> is the ploidy level and <span class="math inline">\(f_i\)</span> is the minor allele frequency of the
<span class="math inline">\(i_th\)</span> marker. If
<code>ploidy.correction=FALSE</code>, it uses the sampling variance
correction as <span class="math inline">\(\sum_i \frac{1}{p}
s^2(m_i)\)</span>, where <span class="math inline">\(s^2(m_i)\)</span>
is the sampling variance of the <span class="math inline">\(i_th\)</span> marker. Both corrections are
equivalent when sampling size goes to the infinity. The default is to
use the sampling variance as the correction (i.e.,
<code>ploidy.correction=FALSE</code>).</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a><span class="co">#Loading the data</span></span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a><span class="fu">data</span>(snp.sol)</span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a><span class="fu">str</span>(snp.sol)</span>
<span id="cb8-4"><a href="#cb8-4" tabindex="-1"></a></span>
<span id="cb8-5"><a href="#cb8-5" tabindex="-1"></a><span class="co">#Computing the additive relationship matrix based on VanRaden 2008</span></span>
<span id="cb8-6"><a href="#cb8-6" tabindex="-1"></a><span class="co"># adapted by Ashraf 2016</span></span>
<span id="cb8-7"><a href="#cb8-7" tabindex="-1"></a>G_VanRaden <span class="ot">&lt;-</span> <span class="fu">Gmatrix</span>(snp.sol, <span class="at">method=</span><span class="st">&quot;VanRaden&quot;</span>, <span class="at">ploidy=</span><span class="dv">4</span>)</span>
<span id="cb8-8"><a href="#cb8-8" tabindex="-1"></a></span>
<span id="cb8-9"><a href="#cb8-9" tabindex="-1"></a><span class="co">#Computing the dominance (digenic) matrix based on Endelman 2018 (Eq. 19)</span></span>
<span id="cb8-10"><a href="#cb8-10" tabindex="-1"></a>G_Dominance <span class="ot">&lt;-</span> <span class="fu">Gmatrix</span>(snp.sol, <span class="at">method=</span><span class="st">&quot;Endelman&quot;</span>, <span class="at">ploidy=</span><span class="dv">4</span>)</span>
<span id="cb8-11"><a href="#cb8-11" tabindex="-1"></a></span>
<span id="cb8-12"><a href="#cb8-12" tabindex="-1"></a><span class="co">#Computing the full-autopolyploid matrix based on Slater 2016 (Eq. 8</span></span>
<span id="cb8-13"><a href="#cb8-13" tabindex="-1"></a><span class="co">#and 9)</span></span>
<span id="cb8-14"><a href="#cb8-14" tabindex="-1"></a>G_FullAutopolyploid <span class="ot">&lt;-</span> <span class="fu">Gmatrix</span>(snp.sol, <span class="at">method=</span><span class="st">&quot;Slater&quot;</span>, <span class="at">ploidy=</span><span class="dv">4</span>)</span>
<span id="cb8-15"><a href="#cb8-15" tabindex="-1"></a></span>
<span id="cb8-16"><a href="#cb8-16" tabindex="-1"></a><span class="co">#Computing the pseudodiploid matrix based on Slater 2016 (Eq. 5, 6,</span></span>
<span id="cb8-17"><a href="#cb8-17" tabindex="-1"></a><span class="co">#and 7)</span></span>
<span id="cb8-18"><a href="#cb8-18" tabindex="-1"></a>G_Pseudodiploid <span class="ot">&lt;-</span> <span class="fu">Gmatrix</span>(snp.sol, <span class="at">method=</span><span class="st">&quot;VanRaden&quot;</span>, <span class="at">ploidy=</span><span class="dv">4</span>, <span class="at">pseudo.diploid=</span><span class="cn">TRUE</span>) </span>
<span id="cb8-19"><a href="#cb8-19" tabindex="-1"></a></span>
<span id="cb8-20"><a href="#cb8-20" tabindex="-1"></a><span class="co">#Computing G matrix with specific weight for each marker as </span></span>
<span id="cb8-21"><a href="#cb8-21" tabindex="-1"></a><span class="co"># in Liu et al. (2020).</span></span>
<span id="cb8-22"><a href="#cb8-22" tabindex="-1"></a>Gmatrix_weighted <span class="ot">&lt;-</span> <span class="fu">Gmatrix</span>(snp.sol, <span class="at">method=</span><span class="st">&quot;VanRaden&quot;</span>, <span class="at">weights =</span> <span class="fu">runif</span>(<span class="dv">3895</span>,<span class="fl">0.001</span>,<span class="fl">0.1</span>), <span class="at">ploidy=</span><span class="dv">4</span>)</span></code></pre></div>
<p>More information about <code>Gmatrix</code> can be found with:</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a>?Gmatrix</span></code></pre></div>
</div>
<div id="ratio-non-dosage-g-matrix-relationship-matrices-using-the-molecular-data-without-dosage-calling" class="section level2">
<h2>Ratio (non-dosage) G matrix: relationship matrices using the
molecular data without dosage calling</h2>
<p>Molecular markers data should be organized in a matrix format
(individual in rows and markers in columns) coded according to a
fraction that represents its molecular information, this can be any
number between 0 to 1. Such ratio can represent the count of alternative
alleles over the read depth for each individual-marker combination
(GBS-like technique). It can be the signal of the alternative allele
over the sum of the signals of the alternative and reference alleles
(GCMS-like technique). It can also be used for family-pool
genotypes.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a><span class="co">#Loading the data</span></span>
<span id="cb10-2"><a href="#cb10-2" tabindex="-1"></a><span class="fu">library</span>(AGHmatrix)</span>
<span id="cb10-3"><a href="#cb10-3" tabindex="-1"></a><span class="fu">data</span>(snp.sol) </span>
<span id="cb10-4"><a href="#cb10-4" tabindex="-1"></a>snp.sol.ratio <span class="ot">=</span> snp.sol<span class="sc">/</span><span class="dv">4</span> <span class="co">#transforming it in a ratio of the minor allele frequency</span></span>
<span id="cb10-5"><a href="#cb10-5" tabindex="-1"></a>Gmatrix <span class="ot">&lt;-</span> <span class="fu">Gmatrix</span>(snp.sol, <span class="at">method=</span><span class="st">&quot;VanRaden&quot;</span>, <span class="at">ploidy=</span><span class="dv">4</span>, <span class="at">ratio=</span><span class="cn">FALSE</span>)  </span>
<span id="cb10-6"><a href="#cb10-6" tabindex="-1"></a>Gmatrix.ratio <span class="ot">&lt;-</span> <span class="fu">Gmatrix</span>(snp.sol.ratio, <span class="at">method=</span><span class="st">&quot;VanRaden&quot;</span>, <span class="at">ploidy=</span><span class="dv">4</span>, <span class="at">ratio=</span><span class="cn">TRUE</span>)</span>
<span id="cb10-7"><a href="#cb10-7" tabindex="-1"></a>Gmatrix[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]<span class="sc">==</span>Gmatrix.ratio[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span>
<span id="cb10-8"><a href="#cb10-8" tabindex="-1"></a></span>
<span id="cb10-9"><a href="#cb10-9" tabindex="-1"></a><span class="do">## it also has the ploidy.correction option</span></span>
<span id="cb10-10"><a href="#cb10-10" tabindex="-1"></a>Gmatrix.alternative <span class="ot">&lt;-</span> <span class="fu">Gmatrix</span>(snp.sol, </span>
<span id="cb10-11"><a href="#cb10-11" tabindex="-1"></a>                               <span class="at">method=</span><span class="st">&quot;VanRaden&quot;</span>, </span>
<span id="cb10-12"><a href="#cb10-12" tabindex="-1"></a>                               <span class="at">ploidy=</span><span class="dv">4</span>, </span>
<span id="cb10-13"><a href="#cb10-13" tabindex="-1"></a>                               <span class="at">ratio=</span><span class="cn">FALSE</span>, </span>
<span id="cb10-14"><a href="#cb10-14" tabindex="-1"></a>                               <span class="at">ploidy.correction=</span><span class="cn">TRUE</span>) </span>
<span id="cb10-15"><a href="#cb10-15" tabindex="-1"></a></span>
<span id="cb10-16"><a href="#cb10-16" tabindex="-1"></a>Gmatrix.ratio.alternative <span class="ot">&lt;-</span> <span class="fu">Gmatrix</span>(snp.sol.ratio, </span>
<span id="cb10-17"><a href="#cb10-17" tabindex="-1"></a>                                     <span class="at">method=</span><span class="st">&quot;VanRaden&quot;</span>, </span>
<span id="cb10-18"><a href="#cb10-18" tabindex="-1"></a>                                     <span class="at">ploidy=</span><span class="dv">4</span>, </span>
<span id="cb10-19"><a href="#cb10-19" tabindex="-1"></a>                                     <span class="at">ratio=</span><span class="cn">TRUE</span>, </span>
<span id="cb10-20"><a href="#cb10-20" tabindex="-1"></a>                                     <span class="at">ploidy.correction=</span><span class="cn">TRUE</span>)</span>
<span id="cb10-21"><a href="#cb10-21" tabindex="-1"></a>Gmatrix[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]<span class="sc">==</span>Gmatrix.alternative[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span>
<span id="cb10-22"><a href="#cb10-22" tabindex="-1"></a>Gmatrix.alternative[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]<span class="sc">==</span>Gmatrix.ratio.alternative[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span></code></pre></div>
</div>
<div id="combined-relationship-matrix---h-matrix" class="section level2">
<h2>Combined relationship matrix - H matrix</h2>
<p>H matrix is the relationship matrix using combined information from
the pedigree and genomic relationship matrices. First, you need to
compute the matrices separated and then use them as input to build the
combined H matrix. Two methods are implemented: <code>Munoz</code>
shrinks the G matrix towards the A matrix scaling the molecular
relatadness by each relationship classes; <code>Martini</code> is a
modified version from Legarra et al. 2009 where combines A and G matrix
using scaling factors. As an example, here we build the matrices using
data from Endelman et al. (2018) (<code>ped.sol</code> and
<code>snp.sol</code>).</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a><span class="fu">data</span>(ped.sol)</span>
<span id="cb11-2"><a href="#cb11-2" tabindex="-1"></a><span class="fu">data</span>(snp.sol)</span>
<span id="cb11-3"><a href="#cb11-3" tabindex="-1"></a></span>
<span id="cb11-4"><a href="#cb11-4" tabindex="-1"></a><span class="co">#Computing the numerator relationship matrix 10% of double-reduction</span></span>
<span id="cb11-5"><a href="#cb11-5" tabindex="-1"></a>Amat <span class="ot">&lt;-</span> <span class="fu">Amatrix</span>(ped.sol, <span class="at">ploidy=</span><span class="dv">4</span>, <span class="at">w =</span> <span class="fl">0.1</span>)</span>
<span id="cb11-6"><a href="#cb11-6" tabindex="-1"></a>Gmat <span class="ot">&lt;-</span> <span class="fu">Gmatrix</span>(snp.sol, <span class="at">ploidy=</span><span class="dv">4</span>,</span>
<span id="cb11-7"><a href="#cb11-7" tabindex="-1"></a>                <span class="at">maf=</span><span class="fl">0.05</span>, <span class="at">method=</span><span class="st">&quot;VanRaden&quot;</span>)</span>
<span id="cb11-8"><a href="#cb11-8" tabindex="-1"></a>Gmat <span class="ot">&lt;-</span> <span class="fu">round</span>(Gmat,<span class="dv">3</span>) <span class="co">#see appendix</span></span>
<span id="cb11-9"><a href="#cb11-9" tabindex="-1"></a></span>
<span id="cb11-10"><a href="#cb11-10" tabindex="-1"></a><span class="co">#Computing H matrix (Martini)</span></span>
<span id="cb11-11"><a href="#cb11-11" tabindex="-1"></a>Hmat_Martini <span class="ot">&lt;-</span> <span class="fu">Hmatrix</span>(<span class="at">A=</span>Amat, <span class="at">G=</span>Gmat, <span class="at">method=</span><span class="st">&quot;Martini&quot;</span>, </span>
<span id="cb11-12"><a href="#cb11-12" tabindex="-1"></a>                        <span class="at">ploidy=</span><span class="dv">4</span>, <span class="at">missingValue=</span><span class="sc">-</span><span class="dv">9</span>, <span class="at">maf=</span><span class="fl">0.05</span>)</span>
<span id="cb11-13"><a href="#cb11-13" tabindex="-1"></a></span>
<span id="cb11-14"><a href="#cb11-14" tabindex="-1"></a><span class="co">#Computing H matrix (Munoz)</span></span>
<span id="cb11-15"><a href="#cb11-15" tabindex="-1"></a>Hmat_Munoz <span class="ot">&lt;-</span> <span class="fu">Hmatrix</span>(<span class="at">A=</span>Amat, <span class="at">G=</span>Gmat, <span class="at">markers =</span> snp.sol, </span>
<span id="cb11-16"><a href="#cb11-16" tabindex="-1"></a>                      <span class="at">ploidy=</span><span class="dv">4</span>, <span class="at">method=</span><span class="st">&quot;Munoz&quot;</span>, </span>
<span id="cb11-17"><a href="#cb11-17" tabindex="-1"></a>                      <span class="at">missingValue=</span><span class="sc">-</span><span class="dv">9</span>, <span class="at">maf=</span><span class="fl">0.05</span>)</span></code></pre></div>
</div>
<div id="covariance-matrices-due-to-epistatic-terms" class="section level2">
<h2>Covariance matrices due to epistatic terms</h2>
<p>Here we present how to compute the epistasis relationship matrices
using Hadamard products (i.e. cell-by-cell product), denoted by
<code>*</code>. For more information please see Munoz et al. (2014). In
this example we are using the molecular-based relationship matrices.
First, build the additive and dominance matrices:</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a><span class="fu">data</span>(snp.pine)</span>
<span id="cb12-2"><a href="#cb12-2" tabindex="-1"></a>A <span class="ot">&lt;-</span> <span class="fu">Gmatrix</span>(<span class="at">SNPmatrix=</span>snp.pine, <span class="at">method=</span><span class="st">&quot;VanRaden&quot;</span>, <span class="at">missingValue=</span><span class="sc">-</span><span class="dv">9</span>, <span class="at">maf=</span><span class="fl">0.05</span>)</span>
<span id="cb12-3"><a href="#cb12-3" tabindex="-1"></a>D <span class="ot">&lt;-</span> <span class="fu">Gmatrix</span>(<span class="at">SNPmatrix=</span>snp.pine, <span class="at">method=</span><span class="st">&quot;Vitezica&quot;</span>, <span class="at">missingValue=</span><span class="sc">-</span><span class="dv">9</span>,<span class="at">maf=</span><span class="fl">0.05</span>)</span></code></pre></div>
<p>For the first degree epistatic terms:</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" tabindex="-1"></a><span class="co">#Additive-by-Additive Interactions</span></span>
<span id="cb13-2"><a href="#cb13-2" tabindex="-1"></a>A_A <span class="ot">&lt;-</span> A<span class="sc">*</span>A</span>
<span id="cb13-3"><a href="#cb13-3" tabindex="-1"></a><span class="co">#Dominance-by-Additive Interactions</span></span>
<span id="cb13-4"><a href="#cb13-4" tabindex="-1"></a>D_A <span class="ot">&lt;-</span> D<span class="sc">*</span>A</span>
<span id="cb13-5"><a href="#cb13-5" tabindex="-1"></a><span class="co">#Dominance-by-Dominance Interactions</span></span>
<span id="cb13-6"><a href="#cb13-6" tabindex="-1"></a>D_D <span class="ot">&lt;-</span> D<span class="sc">*</span>D</span></code></pre></div>
<p>For the second degree epistatic terms:</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" tabindex="-1"></a><span class="co">#Additive-by-Additive-by-Additive Interactions</span></span>
<span id="cb14-2"><a href="#cb14-2" tabindex="-1"></a>A_A_A <span class="ot">&lt;-</span> A<span class="sc">*</span>A<span class="sc">*</span>A</span>
<span id="cb14-3"><a href="#cb14-3" tabindex="-1"></a><span class="co">#Additive-by-Additive-by-Dominance Interactions</span></span>
<span id="cb14-4"><a href="#cb14-4" tabindex="-1"></a>A_A_D <span class="ot">&lt;-</span> A<span class="sc">*</span>A<span class="sc">*</span>D</span>
<span id="cb14-5"><a href="#cb14-5" tabindex="-1"></a><span class="co">#Additive-by-Dominance-by-Dominance Interactions</span></span>
<span id="cb14-6"><a href="#cb14-6" tabindex="-1"></a>A_D_D <span class="ot">&lt;-</span> A<span class="sc">*</span>D<span class="sc">*</span>D</span>
<span id="cb14-7"><a href="#cb14-7" tabindex="-1"></a><span class="co">#Dominance-by-Dominance-by-Dominance Interactions</span></span>
<span id="cb14-8"><a href="#cb14-8" tabindex="-1"></a>D_D_D <span class="ot">&lt;-</span> D<span class="sc">*</span>D<span class="sc">*</span>D</span></code></pre></div>
<p>And so on…</p>
</div>
<div id="exporting-your-matrix-as-three-columns-and-sparse-format-asreml---csv-format" class="section level2">
<h2>Exporting your matrix as three columns and sparse format (ASReml -
csv format)</h2>
<p>That is the lower diagonal matrix formatted in three columns in .csv
format (other ASCII extension could be used as well). In order to do
this, we need to build a matrix, its inverse, and export it using
<code>formatmatrix</code> function. ASReml can invert the relationship
matrix as well, probably more efficiently than R for large matrices
(i.e. <code>solve()</code> function), so no need to invert the matrix in
R if matrix is large. This function has as options:
<code>round.by</code>, which let you decide the number of decimals you
want; <code>exclude.0</code>, if <code>TRUE</code>, remove all the zeros
from your data (i.e., transforms into sparse); and, name that defines
the name to be used in the exported file. Use the default if not sure
what parameter use in these function. Here an example using
<code>ped.mrode</code> data:</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" tabindex="-1"></a><span class="co">#Loading the data example</span></span>
<span id="cb15-2"><a href="#cb15-2" tabindex="-1"></a><span class="fu">data</span>(ped.mrode)</span>
<span id="cb15-3"><a href="#cb15-3" tabindex="-1"></a></span>
<span id="cb15-4"><a href="#cb15-4" tabindex="-1"></a><span class="co">#Computing the matrix</span></span>
<span id="cb15-5"><a href="#cb15-5" tabindex="-1"></a>A <span class="ot">&lt;-</span> <span class="fu">Amatrix</span>(<span class="at">data=</span>ped.mrode, <span class="at">ploidy=</span><span class="dv">4</span>, <span class="at">w=</span><span class="fl">0.1</span>)</span>
<span id="cb15-6"><a href="#cb15-6" tabindex="-1"></a></span>
<span id="cb15-7"><a href="#cb15-7" tabindex="-1"></a><span class="co">#Building its inverse</span></span>
<span id="cb15-8"><a href="#cb15-8" tabindex="-1"></a>Ainv <span class="ot">&lt;-</span> <span class="fu">solve</span>(A)</span>
<span id="cb15-9"><a href="#cb15-9" tabindex="-1"></a></span>
<span id="cb15-10"><a href="#cb15-10" tabindex="-1"></a><span class="co">#Exporting it. The function &quot;formatmatrix&quot; </span></span>
<span id="cb15-11"><a href="#cb15-11" tabindex="-1"></a><span class="co"># will convert it and save in your working directory</span></span>
<span id="cb15-12"><a href="#cb15-12" tabindex="-1"></a><span class="fu">formatmatrix</span>(Ainv, <span class="at">round.by=</span><span class="dv">12</span>, <span class="at">exclude.0=</span><span class="cn">TRUE</span>, <span class="at">name=</span><span class="st">&quot;Ainv&quot;</span>)</span></code></pre></div>
</div>
<div id="relationship-matrices-using-pedigree-data-for-polycrosses---a-matrix-beta" class="section level2">
<h2>Relationship matrices using pedigree data for polycrosses - A matrix
(beta)</h2>
<p>Creates an additive relationship matrix A based on a
non-deterministic pedigree with 4+ columns where each column represents
a possible parent. This function was built with the following designs in
mind. 1) A mating design where you have equally possible parents. For
example, a generation of insects derived from the mating of three
insects in a cage. All the insects in this generation will have the same
expected relatedness with all the possible parents (1/3). If there are
only two parents in the cage, the function assumes no-inbreeding and the
pedigree is deterministic (the individual is offspring of the cross
between the two parents). Another example, a population of 10
open-pollinated plants where you harvest the seeds without tracking the
mother. 2) When <code>fixedParent</code> is TRUE: a mating design where
you know one parent and might know the other possible parents. For
example, a polycross design where you have seeds harvested from a mother
plant and possible polén donors.</p>
<p>The following pedigree has the id of the individual followed by
possible parents. The possible parents are filled from left to right, in
the <code>pedigree</code> data frame: id 1,2,3,4 have unknown parents
and are assumed unrelated; id 5 has three possible parents (1,2,3); id 6
has three possible parents (2,3,4); id 7 has two parents (deterministic
case here, the parents are 3 and 4); id 8 has four possible parents
(5,6,7,1).</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" tabindex="-1"></a>pedigree <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">id=</span><span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>,</span>
<span id="cb16-2"><a href="#cb16-2" tabindex="-1"></a>                      <span class="at">parent1 =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">5</span>),</span>
<span id="cb16-3"><a href="#cb16-3" tabindex="-1"></a>                      <span class="at">parent2 =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">6</span>),</span>
<span id="cb16-4"><a href="#cb16-4" tabindex="-1"></a>                      <span class="at">parent3 =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">0</span>,<span class="dv">7</span>),</span>
<span id="cb16-5"><a href="#cb16-5" tabindex="-1"></a>                      <span class="at">parent4 =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">1</span>),</span>
<span id="cb16-6"><a href="#cb16-6" tabindex="-1"></a>                      <span class="at">parent5 =</span> <span class="dv">0</span>)</span>
<span id="cb16-7"><a href="#cb16-7" tabindex="-1"></a></span>
<span id="cb16-8"><a href="#cb16-8" tabindex="-1"></a><span class="fu">print</span>(pedigree)</span>
<span id="cb16-9"><a href="#cb16-9" tabindex="-1"></a><span class="co">#&gt;   id parent1 parent2 parent3 parent4 parent5</span></span>
<span id="cb16-10"><a href="#cb16-10" tabindex="-1"></a><span class="co">#&gt; 1  1       0       0       0       0       0</span></span>
<span id="cb16-11"><a href="#cb16-11" tabindex="-1"></a><span class="co">#&gt; 2  2       0       0       0       0       0</span></span>
<span id="cb16-12"><a href="#cb16-12" tabindex="-1"></a><span class="co">#&gt; 3  3       0       0       0       0       0</span></span>
<span id="cb16-13"><a href="#cb16-13" tabindex="-1"></a><span class="co">#&gt; 4  4       0       0       0       0       0</span></span>
<span id="cb16-14"><a href="#cb16-14" tabindex="-1"></a><span class="co">#&gt; 5  5       1       2       3       0       0</span></span>
<span id="cb16-15"><a href="#cb16-15" tabindex="-1"></a><span class="co">#&gt; 6  6       2       3       4       0       0</span></span>
<span id="cb16-16"><a href="#cb16-16" tabindex="-1"></a><span class="co">#&gt; 7  7       3       4       0       0       0</span></span>
<span id="cb16-17"><a href="#cb16-17" tabindex="-1"></a><span class="co">#&gt; 8  8       5       6       7       1       0</span></span>
<span id="cb16-18"><a href="#cb16-18" tabindex="-1"></a><span class="fu">AmatrixPolyCross</span>(pedigree)</span>
<span id="cb16-19"><a href="#cb16-19" tabindex="-1"></a><span class="co">#&gt; Your data was chronologically organized with success. </span></span>
<span id="cb16-20"><a href="#cb16-20" tabindex="-1"></a><span class="co">#&gt; Your data was chronologically organized with success. </span></span>
<span id="cb16-21"><a href="#cb16-21" tabindex="-1"></a><span class="co">#&gt; Your data was chronologically organized with success. </span></span>
<span id="cb16-22"><a href="#cb16-22" tabindex="-1"></a><span class="co">#&gt; Your data was chronologically organized with success. </span></span>
<span id="cb16-23"><a href="#cb16-23" tabindex="-1"></a><span class="co">#&gt; Constructing matrix A using ploidy = 2 </span></span>
<span id="cb16-24"><a href="#cb16-24" tabindex="-1"></a><span class="co">#&gt; Completed! Time = 0  minutes</span></span>
<span id="cb16-25"><a href="#cb16-25" tabindex="-1"></a><span class="co">#&gt;           1         2         3         4         5         6         7</span></span>
<span id="cb16-26"><a href="#cb16-26" tabindex="-1"></a><span class="co">#&gt; 1 1.0000000 0.0000000 0.0000000 0.0000000 0.3333333 0.0000000 0.0000000</span></span>
<span id="cb16-27"><a href="#cb16-27" tabindex="-1"></a><span class="co">#&gt; 2 0.0000000 1.0000000 0.0000000 0.0000000 0.3333333 0.3333333 0.0000000</span></span>
<span id="cb16-28"><a href="#cb16-28" tabindex="-1"></a><span class="co">#&gt; 3 0.0000000 0.0000000 1.0000000 0.0000000 0.3333333 0.3333333 0.5000000</span></span>
<span id="cb16-29"><a href="#cb16-29" tabindex="-1"></a><span class="co">#&gt; 4 0.0000000 0.0000000 0.0000000 1.0000000 0.0000000 0.3333333 0.5000000</span></span>
<span id="cb16-30"><a href="#cb16-30" tabindex="-1"></a><span class="co">#&gt; 5 0.3333333 0.3333333 0.3333333 0.0000000 1.0000000 0.2222222 0.1666667</span></span>
<span id="cb16-31"><a href="#cb16-31" tabindex="-1"></a><span class="co">#&gt; 6 0.0000000 0.3333333 0.3333333 0.3333333 0.2222222 1.0000000 0.3333333</span></span>
<span id="cb16-32"><a href="#cb16-32" tabindex="-1"></a><span class="co">#&gt; 7 0.0000000 0.0000000 0.5000000 0.5000000 0.1666667 0.3333333 1.0000000</span></span>
<span id="cb16-33"><a href="#cb16-33" tabindex="-1"></a><span class="co">#&gt; 8 0.3333333 0.1666667 0.2916667 0.2083333 0.4305556 0.3888889 0.3750000</span></span>
<span id="cb16-34"><a href="#cb16-34" tabindex="-1"></a><span class="co">#&gt;           8</span></span>
<span id="cb16-35"><a href="#cb16-35" tabindex="-1"></a><span class="co">#&gt; 1 0.3333333</span></span>
<span id="cb16-36"><a href="#cb16-36" tabindex="-1"></a><span class="co">#&gt; 2 0.1666667</span></span>
<span id="cb16-37"><a href="#cb16-37" tabindex="-1"></a><span class="co">#&gt; 3 0.2916667</span></span>
<span id="cb16-38"><a href="#cb16-38" tabindex="-1"></a><span class="co">#&gt; 4 0.2083333</span></span>
<span id="cb16-39"><a href="#cb16-39" tabindex="-1"></a><span class="co">#&gt; 5 0.4305556</span></span>
<span id="cb16-40"><a href="#cb16-40" tabindex="-1"></a><span class="co">#&gt; 6 0.3888889</span></span>
<span id="cb16-41"><a href="#cb16-41" tabindex="-1"></a><span class="co">#&gt; 7 0.3750000</span></span>
<span id="cb16-42"><a href="#cb16-42" tabindex="-1"></a><span class="co">#&gt; 8 1.0879630</span></span></code></pre></div>
<p>If <code>fixedParent=TRUE</code>, the above pedigree will be
interpreted with the possible parents are filled from left to right
after the known parent, in the <code>pedigree</code> data frame: id
1,2,3,4 have unknown parents and are assumed unrelated; id 5 is
offspring of parent 1 in a deterministic way and two other possible
parents (2,3); id 6 is offspring of parent 2 in a deterministic way and
two other possible parents (3,4); id 7 has two parents (deterministic
case here, the parents are 3 and 4); as before; id 8 is offspring of
parent 5 in a deterministic way and has three other possible parents
(6,7,1).</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" tabindex="-1"></a><span class="fu">AmatrixPolyCross</span>(pedigree,<span class="at">fixedParent=</span><span class="cn">TRUE</span>)</span>
<span id="cb17-2"><a href="#cb17-2" tabindex="-1"></a><span class="co">#&gt; Your data was chronologically organized with success. </span></span>
<span id="cb17-3"><a href="#cb17-3" tabindex="-1"></a><span class="co">#&gt; Your data was chronologically organized with success. </span></span>
<span id="cb17-4"><a href="#cb17-4" tabindex="-1"></a><span class="co">#&gt; Your data was chronologically organized with success. </span></span>
<span id="cb17-5"><a href="#cb17-5" tabindex="-1"></a><span class="co">#&gt; Your data was chronologically organized with success. </span></span>
<span id="cb17-6"><a href="#cb17-6" tabindex="-1"></a><span class="co">#&gt; Completed! Time = 0  minutes</span></span>
<span id="cb17-7"><a href="#cb17-7" tabindex="-1"></a><span class="co">#&gt;           1         2    3     4         5         6         7         8</span></span>
<span id="cb17-8"><a href="#cb17-8" tabindex="-1"></a><span class="co">#&gt; 1 1.0000000 0.0000000 0.00 0.000 0.5000000 0.0000000 0.0000000 0.4166667</span></span>
<span id="cb17-9"><a href="#cb17-9" tabindex="-1"></a><span class="co">#&gt; 2 0.0000000 1.0000000 0.00 0.000 0.2500000 0.5000000 0.0000000 0.2083333</span></span>
<span id="cb17-10"><a href="#cb17-10" tabindex="-1"></a><span class="co">#&gt; 3 0.0000000 0.0000000 1.00 0.000 0.2500000 0.2500000 0.5000000 0.2500000</span></span>
<span id="cb17-11"><a href="#cb17-11" tabindex="-1"></a><span class="co">#&gt; 4 0.0000000 0.0000000 0.00 1.000 0.0000000 0.2500000 0.5000000 0.1250000</span></span>
<span id="cb17-12"><a href="#cb17-12" tabindex="-1"></a><span class="co">#&gt; 5 0.5000000 0.2500000 0.25 0.000 1.0000000 0.1875000 0.1250000 0.6354167</span></span>
<span id="cb17-13"><a href="#cb17-13" tabindex="-1"></a><span class="co">#&gt; 6 0.0000000 0.5000000 0.25 0.250 0.1875000 1.0000000 0.2500000 0.3020833</span></span>
<span id="cb17-14"><a href="#cb17-14" tabindex="-1"></a><span class="co">#&gt; 7 0.0000000 0.0000000 0.50 0.500 0.1250000 0.2500000 1.0000000 0.2708333</span></span>
<span id="cb17-15"><a href="#cb17-15" tabindex="-1"></a><span class="co">#&gt; 8 0.4166667 0.2083333 0.25 0.125 0.6354167 0.3020833 0.2708333 1.1354167</span></span></code></pre></div>
</div>
<div id="amatrix-benchmark" class="section level2">
<h2>Amatrix() benchmark</h2>
<p>It follows a small memory and computational time profiling for the
<code>Amatrix()</code> function. The required RAM was computed based on
the peak of the process for different pedigree sizes (based on
/usr/bin/time -v output). The time profiling was done using AMD Milan
2.95GHz, so it might be an underestimated value when compared with lower
speed processors. Numerator relationship matrices for pedigrees with
less than 20,000 rows can built with low-specs user-end machines
(&lt;8GB RAM) using our package.</p>
<p><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAfgAAAJACAMAAABWh4TIAAAArlBMVEUAAAAAADoAAGYAOjoAOmYAOpAAZpAAZrY6AAA6ADo6AGY6OmY6OpA6ZmY6ZrY6kJA6kNtmAABmADpmAGZmOgBmOjpmOpBmZgBmZrZmkJBmkNtmtv+QOgCQOjqQOmaQZgCQZpCQkGaQkLaQkNuQtpCQ2/+2ZgC2Zjq2Zma2kDq2tv+225C2/7a2///bkDrbkGbbtmbb/7bb/9vb////tmb/25D/29v//7b//9v///+agC3IAAAACXBIWXMAAAsSAAALEgHS3X78AAASf0lEQVR4nO2dC3vbthVAmWye1DTr7GbLusrtuofVbIvVbaIm8f//sREUqUckiiCF18U958sXORFBAveYAEgCYFGBSorYGYA4IF4piFcK4pWCeKUgXimIVwrilYJ4pSBeKYhXCuKVgnilIF4piFcK4pWCeKUgXimIVwrilYJ4pSBeKYhXCuKVgnilIF4piFcK4pWCeKUgXimIVwrilYJ4pSBeKYhXCuKVgnilIF4piFcK4pWCeKUgXimIVwrilYJ4pSBeKYhXCuKVgnilIF4piFcK4pWCeKUgXimIVwrilYJ4pSBeKYhXCuKVgnilIF4piFcK4pWCeKUgXimIVwrilYJ4pSBeKYhXCuKVgvhcKIpRLhGfCUU1TqbNtquieKx2P75OzBKEwIP47cd1tXr4L+KTxpP4avPV14hPGg9t/OqhNl++RXxO0LlTio34zbyuRjjh88JC/O6ndVUumpa+TdPhM2MwitEuLDt3m++vXM4hPhnGq7BJUZqKfvno4GjgCT/ifaQFl0wwgfgc8CN++9T05C679YhPhCkirNIsF86OBx7wJt5DWnDHJA+Ilw/idTJNA+LFg3idTLSAeOkgXidTJSBeOIjXyWQHiJcN4nUyXQHiRYN4ndxhAPGSQbxOYslDfGQQr5O74o94uSBeJ/eFH/FiQbxO7ow+4qWCeJ3cG3zECwXxOrk79ogXyf2hR7xIEK8TB5FHvEQQrxMXgUe8QBCvEydxR7w8EK8TN2FHvDgQrxNHUUe8NBCvE1dBR7wwEK8TZzFHvCwQrxN3IUe8KBCvE4cRR7wkEK8TlwFHvCAQrxOn8Ua8HBCvE7fhRrwYEK8Tx9FGvBQQrxPXwUa8EBCvE+exRrwMEK8T96FGvFIQrxTEKwXxSkF88hSFj0AjPnWKykukEZ86iFcK4rVCGw8OQbxSEK8UxCeOryAjPm28xRjxaYN4nfgLMeKTBvE68RhhxKcM4nXiM8CIT5gA4svC8PZ1SlrwhNf47ne+fVpMTgue8BveVvyHcef6aVrwRAjx1fJxelrwgufodlU9bXxqBBEfPi0M4Du4iE+UQOJ3z3VN/7CelBY84D22+wPsnk3nbnXDfHml94d4b/gP7enlXM9FXW/PD/HeCCX+9hm/mS8444MSILJ2bfzy4d+ID0g48UNs5ogPR4jA2h6Dqj4gocRvP3y+deeOzl1ggsTV5iBfdu6KDm+50k1A8Tcv5yo6d0EJE9amqn9qz+D+bj2du2AEiirP41MjqPjwaaGHUEE9jLlbVKs3L1c3oVcfkrDit98a55t3PRX+8vqIPMS7J1hMu3v1Rm058rks4t0TWPy+Oh858grx7gkXUjp3KREwoohPifDimUmTAiEDykyahIggnjt3CRA0nsykSYcI4plJkwBhw0mvPhkQr5PA0aSqT4TQwTw93mpkDw/xDokpfuxFHeLdETyWpwcsqeqjEUl828aPvH2HeGeED+V+XP2U+3aId0gs8Z+5ZRuVCJFsDrlsh1fTxschRiB5SJMA8cSHTwtHosQR8fFBvE7ihBHx0YkqvuQGTiwiRbE5bFnMzMfueZx6xLsgovjtx8MMmk9jLusQ74BYQWzEf3f4J+K1QFWvFDp3SuFyTimIj0fUdcPahzRNv373I0/nAlJUMUPYHnkzn1XLnqVQhtLCNJIQv18FZ2pamEIS4jnjI0AbD+GhV6+UdrAlU6i0wRkfi8jRY7BlJGIHr13g8IeRHfqTtDCJ2MFjmnQcoseONj4K8UN32sYzTToY8UPXXM4Nv6GiNy1MIYHI0auPQAqBo42PQAqBa/OwolcfjiTi1lb1376Us2o1m5IWRpJG2A5t/P7PhLQwkjTCdrhzV//pfSfNzbQwjkSi1g3EePdaFgXr3AUgkajRqw9MKkE7m1BBr947ycTs0KufnBbGkEzMuHMXlHRC1t3A4Q0VIUgoYty5C0lCEaOND0hKAaOND0dS8aKND0dS8WLMXTDSChd37kKRWLQQH4rEonWSnRVVvUdSC1aXn7Ioxs6STq4sKZNcrJoMbea19SULI3gkuVjtZ8uaKh7xHkkvVE2OzNXcAvH+SDBShywtaeP9kWCkTrK0e6ZX74cUA8V1vH+SjBPi/ZNknI69+mlpYZA0w9QufsRjWW8kGqUmW7xp0iOJRomBGJ5JNUh07jyTapDafO2exy+IkWyZUiLZGLWTJp/N0KsVS6G4Jt0QsfiRV9INEWe8TxKOEG28R1IOEL16j6QcIMT7I+n4IN4baYcH8d5IOzzt5dzfzN+8k8YliUfn5HJuxVuoFHG4nPvtvHfe5GrWjMbsSwsSOQ627PW+/bj+5eVaO4B4wTQDMZ5uPo/f/bQuF+bva2nhGlFfJWiHTQY38+LqBKvkCxeLoko/OFPyV3Q4z00mCBJ/c/Gj5oy/8l3qZYuGHPE3ly03rXu52L9+9kpauERAbXh4Ht+/bLlRvvmeXn1etDdwbi5bXpqK/srlHuIF08pj2XJt8JDGNUKignjHSAnKcejVw3/GrmsqpYxBkRKUw9O5zft1yWDLuxETk8PlXC2e4dV3Iyckp2c8w6vvRVBEGF7tEkERoVfvEEkBQbw7RMXjtKof+WpZWQUNgKh4nM2d46XC9yArHMyWdYWwaHQDMWYVZ/xdSAtGe8bfHm55My00SAsGvXo3iIsF4p0gLxSId4K8UCDeBQIjgXgHSAwEAzHuR2QcGIhxPyLjwECMu5EZBgZi3IvQKDAQ416ERoFe/Z1IDQLi70NsDNqMl7yhYhJyQ3CYJj05rWrkhuB0IMa0tJoRHIFuIMbIibKnafUiOQAWq17dTKsayQGgVz8d0eVnsOVkZBf/rKrnlu0IhJeeXv1UhJeeNn4i0guP+GmILzvipyG+7PTqJyG/6PTqp5BByenVTyCHgtPGTyCHgncPabhXb08W5T48j+9ftnwgLYjk0Mb3L1s+kBZE0g6vvrls+e20IJJWHsuW2yHg1ROW0KsfQ1FlU2rEjwHxd6eVCeLvTisU2vh700JkzmbSXL4w2iYtSKSRV+6Xsd09j1OPeME0j2WPL5H8xAKHveRVXNp4WzIrbVOczbyYlaMfzuUWitvkVlhTnt3zolrV1m8tflReuZ2bWyxukV1Zmzb+w2u1eb/uG4fTjsy6rA+yC8YNsiurhfi6JVhoP+PzK6qN+KpaPvxbtfgMS2o5TXoz1yw+x4JOKVPR4Tw3aZJlOW0KtZqZOuHypl6WAbkkz2LuS7WamXG2PTX99uP6l5dq96PSXn2mpWyKZdYyXT32jbLd/bQuF+bvq2mzJ9NSNp07s9hZLb5vsOVmXrfnby4XRMs0JOfkWsjucq6qmDR5jWzL2N2yNbBe/QX5FrF9Hm/Ml1dq8+G0WZNxCdsJFc/jZ85lHZY9OReQ5/H9ZF2+pnP33eGfjMA5IevyMeaul7yLxyjbPvIuHW18H1kXrkJ8HzmXrQHxV8m4aC2Iv0rGRWtB/DXyLdkBxF8h24KdsH9Iw6tJTsm1XGfsB2KMXv7mmDY/Mi3WF3SlNCscjpWfZ4TyLNUFp8VcUtVnWqgrdOVcjr9jm2eMsizUFbo2fqz0Q9rMyLFMV6FXf0aGReqB6/hT8itRL4hXyom8FVW9Ijp55dUpE3ZpQSDdGjhvXpa9HftV8fCP3CdN6pn723KcSdMrfvtxXT5mPmkyo0VqLWlK28yCRnzsTATlUNplfxuvoKpXLN7cx7nRqz9d/CjHFTHyKo0FZ8X939VJkyx3liNdr77W2nvGs9xZhuzv1ZvVMMobr5bNfLmzbAoygu5yrq7Pbz2gy3q5s1zKMYqD+G/H3rfLJmCZFGMkB/ET3iedScQyKcZIEJ9HKUZjuaRpb1rxZFGICWh/Hp9DGSaBeKUoF59BESaiW7z8EkxGtXjxBbgDxCtFs3jp+b8LxeKFZ/9O9IqXnfu7UStedOYdgHilaBUvOe9OUCpecNYdoVO83Jw7A/FKUSlebMYdolG81Hw7RaF4odl2DOKVok+8zFw7R514kZn2gE0cmqVuM5kfLzHPXrAIxPbjulo9/DcL8QKz7AlL8dXmq68RnxNWVb2ZRlvmMD9eXo69MSUUYlfEEJdhj9h17h7+nMMaONLy6xW7Nr6c5dCrl5Zfr9h27jIQLyy7nrGJRll37nbP0qt6Wbn1jpo7d6IyGwDEK0WLeEl5DYIS8YKyGggl4uFLEK8UxCsle/ECHykEIXfxRSUjn8FBvFIQr5TcxdPG95C9eLhOzuJTz19U8hVPFX+TXMWjfYA8xaN9kCzFJ5uxhMhQPKe7DdmJR7sdmYlHuy15iU8vR8mSk3hO9xHkIx7to7CaOzczLyhLfEJFUpkRgN0Uql9eEp9Cxek+FouA7X5alwvz94S0YUD7eGxCtpkXRfHm8qXDiYQb7VOQvzBCGrkQRz69ehiFTefuqeeNw3HFp1LjCMUqeMvLSznrtL5gFOV9iK3qEX8fiFeKWPG08fchUTzGHSBPPNqdIE082h0hSjzNujsEice6S6SI52R3jAzxWHeOBPFo90Dy4qnj/WA15q54+EekMXdY94XdmLvy7C1UwQZioN0fU8SPSDuJ9lcK7T5JsKrnuVsIEuzcIT4EiFdKUuLbVp0ruAAkIz6Z4dpKSEM80oMTXzynehTizqRBejQinvFIj0lQ8UfVnOqxCSn+cIGO9PhEEQ/x8Sb+tC7vnrrceTxwiDvx5632UfLJ/9Owp4ONiWVz7fZ4O+0XDThnd+LYPY+vqsGBGF+YRnziTBHfu6Ozup5qPWnsBmJcr+q/2BOmJYEspSBeKYhXCuKVgnilIF4piFcK4pWCeKUgXimIVwrilYJ4pTgUX0Bcookf3rPl0ew2c7qzKJtFydqkrcfsCvFJHnPa1mN2hfgkjzlt6zG7QnySx5y29ZhdIT7JY07besyuEJ/kMadtPWZXiE/ymNO2hmxAvFIQrxTEKwXxSkG8UhCvFMQrBfFKQbxSEK8UxCvFnfjV6btLds/Fm5fLDer/nt3ey/apePs6uFmzwfDOqmrzfj28WZNZq80eLPZm1g95tCmBzc7qcFgUtIusTUQ6nIlfnr20Zvvd1Q3Kx2p5+W6bU+oyrB4HN6s3+PlleGd1JB7Ww5s1mbXL2sLioFX16dWmBDY7q49ZDm52iKxV5lpcid/+8+yM38zrX+jLDX55MTkcoFxYbLb78dViq9XvPq6HN2syO7zZpz9abdZIHdysET+8MyN+NrDZMbJW4W3xVNXXx998f7nBzxY5235cD2+2fZpZ7Gzz/j82O2syO7zZ0pxXNiX49GqRt1XxaLFVXdX/6pvBzbrI2mSuw5P4qrrMgt0Zv/3walUx2Px+N82tXS1jsVnt0+qk2nxTDZdgVZ/xzrKW0BnftE2XGww3Qpt3rxbtbbMrmxZt/7JEFw2p9WZN7K1KMLM6pkXTvYrYxl/26i/6l1a9+qXLPnEj3qq7btMjtj2oCcTgZnZXL5ZZi9qrB1kgXimIVwrilYJ4pSBeKYhXCuKVgnilIF4piFcK4pWCeKUgXimIVwrilYJ4pSBeKYhXCuKVIlj8Zm7W6j4bTrx7XjTDdIfTXtlq9fjl/3b/3j4100POP27v//3QFrGRLN5o2X57OkWvFj99dxeuyuLta7fX1eyLjwFWtuPbYyFdvNHQDFVuzsRf/745481Pf3jZfP37t6/H7/YSa5vmp3orM5B7dvLFcmH+d/P+z0U7C3D55u/7M97M8ai/Ov8wGTgcwORi89VLVT6s9/tv0iSNdPEm3MvH5hSsP8qiEW9+evOymS+qk+/2Z6kRUv+0/6X54eX8CyN+/mj0nRxhXxfUNcv5h/n/9gD19mZiTGHme7b7r+xnNkRCsvimjV80Fup4m5Dv2/jmpx9emnO//W7/UR2ahkZpbev4hfFpxL97PXYA2h/ML0K94flH+7VJvP9t+Pkvs/qgh6Yn9bpesvhj16so3jSe6/Osqa7XB/H779qPJtXc/ND4Nafu4Qvjs0d87xnfbmt8f/j84V/vPptaY77fH+K9cRD/4dASX57xx++O1I7r70xFf/LFjTO+t40/nvHV8k/f7H74y6zbP+I9cvCzb2Yv2/imCjh+17Tc5u+9+L2ZwxddG39F/O75cd+dP/3ovm53UBb7Vr7dP228Rw5+6vra1K6755Ne/W/2Z3z3XftRNbMym17956fmLsDJF4ur4tv9XbuO7w7wtu1kluaHJb36qNjdxzlL0XPP5dMkg6nX9FmKr0/9y6WXhrnuavfXKVngzh0kCuKVgnilIF4piFcK4pWCeKUgXimIVwrilYJ4pSBeKYhXCuKVgnilIF4p/weQKgRQ6AUIgAAAAABJRU5ErkJggg==" style="display: block; margin: auto;" /><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAfgAAAJACAMAAABWh4TIAAAApVBMVEUAAAAAADoAAGYAOjoAOmYAOpAAZpAAZrY6AAA6ADo6AGY6OmY6OpA6ZrY6kJA6kNtmAABmADpmAGZmOgBmOjpmOpBmZgBmZrZmkJBmkNtmtv+QOgCQOjqQOmaQZgCQZpCQkGaQkLaQtpCQ2/+2ZgC2Zjq2Zma2kDq2tv+225C2/7a2///bkDrbkGbbtmbb/7bb/9vb////tmb/25D//7b//9v///8lY2dcAAAACXBIWXMAAAsSAAALEgHS3X78AAAR20lEQVR4nO2diXbbuBVAmbSqnWUqx1N3WjnpTGtNpo01iyWL//9pJSGKpixRBCksD3j3npzICUESeNdYuAAqSlBJETsDEAfEKwXxSkG8UhCvFMQrBfFKQbxSEK8UxCsF8UpBvFIQrxTEKwXxSkG8UhCvFMQrBfFKQbxSEK8UxCsF8UpBvFIQrxTEKwXxSkG8UhCvFMQrBfFKQbxSEK8UxCsF8UpBvFIQrxTEKwXxSkG8UhCvFMQrBfFKQbxSEK8UxCsF8UpBvFIQrxTEKwXxSkG8UhCvFMQrBfFKQbxSEK8UxCsF8UpBvFIQrxTEKwXxSkG8UhCvFMQrBfFKQbxSEK8UxCvFRvyyqJl7zwsExEL8891T9ff2y6P3zEA4EJ8LVaM8KrlFmhVNvXyKctyAbcrgrtgzYV/whA/xq7q6n2jqES8ID+LrPn41+wPxsnHfx5vB3ebdB8TnhFVTP6vMr98iPicukYf4hEG8UhCfB6NdID4PEK+T8SoQnwWI18kEE4jPAcQrBfE6mfSINfD5wAOIVwridTLJA+LTB/E6maYB8cmDeKUgXicTLSA+dRCvk6kSEJ84iFcK4nUy2QHi0wbxSkG8TqYrQHzSIF4nFxhAfMogXimI10kseYiPjG95q2L2S1EsnJ4XLuei+NsthbJm8SOBIF4pvsXT1MvksvCzwGGyhBfv5sxwId7FP9+Y6s1yZ7K4MPpWuy+P+3cHp4aLCCHez6nhEi4NPuITBfFKQbxOLo494tME8Tq5PPSITxLEKwXxOnEQecSnCOKVgniduAg84hME8TpxEnfEpwfilYJ4nbgJO+KTA/FKQbxOHEUd8amBeJ24CjriEwPxSkG8TpzFHPFpgXiduAs54pMC8UpBvE4cRhzxKYF4pSBeJy4DbrXq1VW9GgqrXkUnsPjnu6dfH1jnLj5O421xsO2PT+tF/bfXjMAgocWXm+uiKN48+M0IDBJc/NE+LHAYA7fhZlSfDMHFs8ChCBxHmwUOUyGGeA/7wlgQrxPXwUZ8IiBeKZ7Er3sG7kHzAv04j/XugM83p8ftNvtCCHyJvx1X17v7QgDch7o54nI+fV/wjy/xvTfnAucGgsGoXimIV0ojb3tftfSzo1ctrPaFFNnJ297Xg7vVSPOID4KfFx+6l3NjL+oQH4Ki9BJparx0vIqnj5eLX/HB9wVrfPbx4feFyNTynm+/cedOG9R40fgLMZdzkvEYYdPU3zQTJLick4XPAPM8Xi5e40sfLxa/4W3fuVuUqxPzIm32BS94jm7T1H+qnW/eM7gTg+/g7u/V1y9brhncicF7bJsTmIH9yPs3iPdIKPHB94Wz+A8t4iUSILLMpBFIiMAyk0YeQeJqe+dufWLGBeK9ECasFjNpWAolKIGiajOTZnO9oMaHIlRQ7c6znP2G+CAEi6nliTbXiA9BuJBOmTTJAoe+CBjR7qlWI+dKI94xIQPaPVfPRR2j+kAEjWf3ZOu+pp4FDkMQNpwHffzI23eId0ngaO7eq5/yxh3inRI6mM2ECl62jE0M8eWyuTzj6Vw0gseS16tFED6UvIghgQiRRLwAYgQS8fGJEkfERydOGA/eueMGTgQiRdGcdl1c1R/b+3HqEZ8w5gbOXTuD5uuYyzrEJ4wR/0P7T8QHJOobDTT10fC0jtmIs5cM7mIgQnzwfQHxaonfxzfj+u0Xns6poZG3ub4qlyyFoohW3nrs0A7xSUONVwp9fCwiR49RfSwEiGf16gjEDh41PhKxg8fLlnGIHrtmgcPPIwf0nX1hCtFjN2Wa9OG+MIH4oaOPj0L80HX7eL6hIhQCImcu5/iGisAIiByj+ghICBx9fAQkBK7Jw+rcqL7aOD91I19C/lNERNyapv7Tw/qqXF2dTFI/wVnN/kC8K0TEre3jd39OJTGP7jbvPiDeDTLC1t65q/70fSeN+XrxEysjyShBcsgI2/5FjPeP67ont9uHBQ4vQEjUGNWHRkjUDiZUnB7Vs8ChU4RErR3Vn0vEAofukBI07twFRkrQ9jdwRq5f3N0XRiAmZjZ37gb2hRGIiZlVH392XxiBnJDRxwdFTsjo40MiKGK8cxcSQRHjzl1AJAUM8QGRFLBOXlY09X4RFa99ZtZFMXaWtKyCJICoeJnMbK4r6z035If2BVtkhWs3W7Zu4hHvGVnhMrmpr+YWiPeLsGi12VnSx/tFWLQ62dneM6r3iLBocR0fCGnBQnwgpAUL8WEQFyvEh0FcrJoMbe+L2e9j38YQVxi5yAtVM5Pmfr75+LRmfrwv5IWqfQOnEs+KGL4QGKlujV9R4z0hMFIvffzolVAkFkcmEgPFqD4AEgOFeP+IjJPFpMmhfWEAkXFiQoV3ZIaJCRXekRkmJlT4RmiUDla2pI/3gNAoMar3jdAodfv4vp5+aZqD485AaJFkITVIB0396Vt3vV9RJbVMopAaJItRPeIvQGyMbDK2oqmfjNgYTckYCxxaIzdEjOq9IjdEFjljgcPJCI7QwUOavklULHA4EcERMllbF2al+u39uK8SF1wsGUgOkLmOv2uv37+OuWkruVwikBwgBnf+EB2f/dO52dOqv48/vy/0IDo+7YsY1Z++b6g4vy/0IDs87S3bqs4j3imyw7Nv6os3D2uaepcIjw6DO18Ijw7ilcJsWaUwW1YpzJZVCrNlfZDAuwrMlvVAUcoPDqN6D6Qj/vzr1ef3hSNSET/wevXZfeEUyfTxTJpUB328Upoaz7dQOSORqHSzOXaudCJFDEsqQenmk1H95SQTk25G1zT1l5JOSA76eF7EuJCEIsKo3iEpBQTx7kgqHqxz54y0wrHv40d/91iZWkm9k1Y4uGXrisSi0WR3yTp3F5JaMLhl64bkYsGo3gnphQLxLkgwEuZFjNtvNPWXkGIgqPGXk2QcuIFzMWmGoZ0ffybNqpj9cuoJTpoldk2iUbBb0nQ97y5pygKHHVINwn5+/JkbOEfiX+2rmmRj0C6McKaPp6nvJd0Q2PTx5/fVTMIR4CHNBaQcAIs+fmhftSRdfh7STCfp8nPnbjJpF9/cq/+h/Sdr2VqTeOlN9lm9ejypF/7gXj3v1VuTfNnp4yeRftERP4UMSo74CeRQcMQrBfFK4Q2cUeTzEgJTqMaQwjpmlvB0bgzZiWcKlR3ZiefpnCW59fHh94XIIF4pNPVKYYFDe7IqLwscWpNXcVng0Jq8issCh7ZkVlpG9ZbkVlhu2dqRXVl3Bdp+ZgrVebIrK9fxVuRXVPp4GzIsKeItyLGgiLcgx4Iifpgsy4n4QfIsZlMqvlS4l0xL2VzH39cP5vga8VNkWsrunTuezp0g10La1PjVVX2DR+eqV9mW0aKPf757+vVB6QKH+RbRomTbH5/Wi/rvCfumTr5FtOnjN9dV7X5z/Bgn36jsybiEZg2cm6bpZlR/SM4F5Hl8P1mXjzt3/WRdPsT3knfxEN9H3qVDfC95l65zA2f2+9i1y7MOTdaFKzu3bDcfn9ZczrXkXDZDezlXiechTUvGRWvo1ngey7ZkXLQGXsQ4Rb4la2FUf4JsC9aBde6OybVcBzSDO76FqkOu5TqAhzRHZFqsVzSl5FuoWvIs1RF78fTxDVkW6gT08a/IslAnoI8/JMcynYQ+/oAMi9QDCyN0ya9EvXDnrkt+JeoF8R2yK9AZzOvVt99o6rVBjVfKrsZPuZhDfNIgfoeKKaBdEG8oyqyKY8HB3Dm9gzul4qnxiB+7by7Qx4/cFxKF6/hcijESxOdRitFoF6+ta29RLj6DIkxEt/j0SzAZ26KvT7yik3zYki/ABViUvff1nNTjlnr+L8Km8JvrxUGNz2Nly8Szfyl2pV/OfsutqU868w6wLP/mOjPxKefdCUpH9Qln3REqxSvv3g0axaeab6coFJ9oth2jT3yauXaONvF07w3KxCeYZU/oEp9ejr2hSnxyGfaIJvGp5dcresQzrDtAjfikMhsALeJTymsQtIiHVyBeKdmLT/1FIV/kLl7fbEhLEK8UxCsld/H08T1kLx5Ok7N46fmLSr7iaeLPkq140ZkTQK7iJedNBHmKp5kfJEvxYjMmiBzFS82XKPITTzNvRXbiRWZKILmJl5gnkeQlnmbemqzEi8uQYHISLy0/oslHPM38KLIRLyozCZCLeEl5SQKbgK2KYl5uvwhe4JBmfjQ2K1vePZWr2R8v4sUtcCgmIwlhKb7cvPsgtsZLyUdSWDX1s8r8WupatnIanqRIfnAnIhMJkrx4mEa64gUNLlMkWfFMkbkMxCsF8UpJVjx9/GWkKB7jDkhPPNqdkJh42ndXpCQe6w5JRzzWnZKIeCq7a5IQj3X3yBdPZfeCcPFY94Vo8Vj3h0TxTTVHu08EiufxSwgQrxTEK0WgeIbyIRAlHuHhkCOeeh4UKeKxHhgR4qns4YkvHutRiCwe67GIKR7rEYknHutRiSSeyh6bKOKxHp+g4qnncggpnqcvgvAmvlu7929WXHg+cIgv8S+SX34DEC8Id+IPO/CTkunj5WBjYmkWtZuf37c1TbOeBBZyzDp33ZUtTy5w+No0tVs2U8T3HgjV6WDjatXT1L86ElU8JZClFMQrBfFKQbxSEK8UxCsF8UpBvFIQrxTEKwXxSkG8UhCvFIfiC4hLNPHDR7Y8m10ypweLkixK1ialHnMoxIs857TUYw6FeJHnnJZ6zKEQL/Kc01KPORTiRZ5zWuoxh0K8yHNOSz3mUIgXec5pqSEbEK8UxCsF8UpBvFIQrxTEKwXxSkG8UhCvFMQrBfFKcSd+tej8Y3tfvHk4TlD999X5ozzfFG8fB5OZBMMHK8vNx6fhZCazVslmFker1w+Z25TA5mBVOCwKuo+sTUT2OBO/LLrin384mWA9L5eLoy1dqjKs5oPJqgQ/PwwfrIrE7Gk4mcmsXdYWFicty6+PNiWwOVh1zvVgsjayVplrcCX++b8HNX5zXf1CHyf49aHO4QDrhUWy7ZdHi1Srv949DSczmR1O9vXvVsmM1MFkRvzwwWrxVwPJXiJrFd4GT019df7NP48T/GyRs+e7p+FkzzdXFgfbfPzd5mAms8PJlnW9sinB10eLvK2KuUWqqqn/03eDyfaRtcncHk/iy/I4C3Y1/vn20aphsPn9Nt2tXStjkazyaVWpNt+VwyVYVTXeWdYE1XjTNx0nGO6ENu8fLfpbcyibHq1qPhx1pNbJTOytSnBldU6LrnsVsY8/HtUfjS+tRvVLl2NiI95quG4zIrY9aR2IwWR2Vy+WWYs6qoe0QLxSEK8UxCsF8UpBvFIQrxTEKwXxSkG8UhCvFMQrBfFKQbxSEK8UxCsF8UpBvFIQrxTEKyVh8Zvreq3ug9eJt/cL85ru8L4nUq3mr/93/+/nGzM95PDj/PE/DqWITcriay3Pn7pT9Crx0w935GpdvH3cH3V19epjgJXt++2xSF18rcG8qmxq4p+/NzW+/ulvD5sP3799fNm2k1jZrH+qUtUvcl91NiwX9f9uPv6raGYBLt/8Z1fj6zke1abDjzoD7QnqXGzePZTr2dPu+GYf0aQuvg73cm6qYPWxLoz4+qc3D5vrRdnZtqultZDqp90vzeeHww21+Ot5ra9zhl1bULUshx/1/zcnqNLXE2OKer5nc/zSfmZDJFIWb/r4hbFQxbsO+a6PNz99fjB1v9m2+yjbrsEorWy9bKh91uLfP74MAJof6l+EKuHhR7O53nn32/DzT1fVSduuR3pbn7L4l6FXUbwxnqt6Zprrp1b8blvzYfa6rn8wfuuq226offaI763xTdra9+232/+9/1a3Gte74yHeG63427YnPq7xL9teqBxX2+qGvrPhTI3v7eNfany5/Md3288/Xe2Pj3iPtH523exxH2+agJdtpueu/96J35lpN+z7+BPit/fz3XC++7Hf3BxgXex6+eb49PEeaf1U7XXdum7vO6P6v+xq/H5b81GaWZlmVP/txtwF6GxYnBTfHO/Udfz+BG+bQea6/mHJqD4qdvdxDvbouefydZJB6S19luKrqn+89NIwp11t/z0lC9y5A6EgXimIVwrilYJ4pSBeKYhXCuKVgnilIF4piFcK4pWCeKUgXimIVwrilfJ/ICnjFiO89CkAAAAASUVORK5CYII=" style="display: block; margin: auto;" /></p>
</div>
<div id="bibliography" class="section level2">
<h2>Bibliography</h2>
<p>Amadeu, RR, et al. 2023 AGHmatrix: genetic relationship matrices in
R. Bioinformatics 39, 7.</p>
<p>Amadeu, RR, et al. 2016 AGHmatrix: R package to construct
relationship matrices for autotetraploid and diploid species: a
blueberry example. The Plant Genome 9, 4.</p>
<p>Ashraf, BH, et al. 2016. Estimating genomic heritabilities at the
level of family-pool samples of perennial ryegrass using
genotyping-by-sequencing. Theoretical and Applied Genetics 129,
45-52.</p>
<p>Cockerham, CC. 1954 An extension of the concept of partitioning
hereditary variance for analysis of covariances among relatives when
epistasis is present. Genetics 39, 859–882.</p>
<p>de Bem Oliveira, I, et al. 2019 Genomic prediction of
autotetraploids; influence of relationship matrices, allele dosage, and
continuous genotyping calls in phenotype prediction. G3: Genes, Genomes,
Genetics, 9(4), pp.1189-1198.</p>
<p>Endelman, JB, et al. 2018 Genetic variance partitioning and
genome-wide prediction with allele dosage information in autotetraploid
potato. Genetics 209, 77-87.</p>
<p>Hamilton, MG, et al. 2017 Computation of the inverse additive
relationship matrix for autopolyploid and multiple-ploidy populations.
Theoretical and Applied Genetics 131, 851-890.</p>
<p>Henderson, C. 1976 A simple method for computing the inverse of a
numerator relationship matrix used in prediction of breeding values.
Biometrics 32, 69–83.</p>
<p>Kerr, RJ, et al., 2012 Use of the numerator relation ship matrix in
genetic analysis of autopolyploid species. Theoretical and Applied
Genetics 124, 1271–1282.</p>
<p>Legarra, A, et al. 2009 A relationship matrix including full pedigree
and genomic information. Journal of Dairy Science 92, 4656–4663.</p>
<p>Liu, A, et al. 2020. Weighted single-step genomic best linear
unbiased prediction integrating variants selected from sequencing data
by association and bioinformatics analyses. Genet Sel Evol 52, 48.</p>
<p>Martini, JW, et al. 2018, The effect of the H<span class="math inline">\(^{1}\)</span> scaling factors <span class="math inline">\(\tau\)</span> and <span class="math inline">\(\omega\)</span> on the structure of H in the
single-step procedure. Genetics Selection Evolution 50(1), 16.</p>
<p>Mrode, RA. 2014 <em>Linear models for the prediction of animal
breeding values</em>. Cabi. 3rd ed.</p>
<p>Munoz, PR, et al., 2014 Unraveling additive from nonadditive effects
using genomic relationship matrices. Genetics 198, 1759–1768.</p>
<p>R Core Team, 2016 <em>R</em>: A Language and Environment for
Statistical Computing. R Foundation for Statistical Computing, Vienna,
Austria.</p>
<p>Resende, MF, et al. 2012 Accuracy of genomic selection methods in a
standard data set of loblolly pine (<em>Pinus taeda</em> l.). Genetics
190, 1503–1510.</p>
<p>Slater, AT, et al. 2014 Improving the analysis of low heritability
complex traits for enhanced genetic gain in potato. Theoretical and
applied genetics 127, 809–820.</p>
<p>Slater, AT, et al. 2016 Improving genetic gain with genomic selection
in autotetraploid potato. The Plant Genome 9.</p>
<p>Su, G, et al. 2012 Estimating additive and non-additive genetic
variances and predicting genetic merits using genome-wide dense single
nucleotide polymorphism markers. PloS one 7, e45293. <a href="https://doi.org/10.1371/journal.pone.0045293" class="uri">https://doi.org/10.1371/journal.pone.0045293</a></p>
<p>VanRaden, P. 2008 Efficient methods to compute genomic predictions.
Journal of dairy science 91, 4414–4423.</p>
<p>Vitezica, ZG, et al. 2013 On the additive and dominant variance and
covariance of individuals within the genomic selection scope. Genetics
195, 1223–1230.</p>
<p>Yang, J, et al. 2010 Common snps explain a large proportion of the
heritability for human height. Nature genetics 42, 565–569.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span>
<span id="cb18-2"><a href="#cb18-2" tabindex="-1"></a><span class="co">#&gt; R version 4.2.3 (2023-03-15 ucrt)</span></span>
<span id="cb18-3"><a href="#cb18-3" tabindex="-1"></a><span class="co">#&gt; Platform: x86_64-w64-mingw32/x64 (64-bit)</span></span>
<span id="cb18-4"><a href="#cb18-4" tabindex="-1"></a><span class="co">#&gt; Running under: Windows 10 x64 (build 19044)</span></span>
<span id="cb18-5"><a href="#cb18-5" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb18-6"><a href="#cb18-6" tabindex="-1"></a><span class="co">#&gt; Matrix products: default</span></span>
<span id="cb18-7"><a href="#cb18-7" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb18-8"><a href="#cb18-8" tabindex="-1"></a><span class="co">#&gt; locale:</span></span>
<span id="cb18-9"><a href="#cb18-9" tabindex="-1"></a><span class="co">#&gt; [1] LC_COLLATE=C                          </span></span>
<span id="cb18-10"><a href="#cb18-10" tabindex="-1"></a><span class="co">#&gt; [2] LC_CTYPE=English_United States.utf8   </span></span>
<span id="cb18-11"><a href="#cb18-11" tabindex="-1"></a><span class="co">#&gt; [3] LC_MONETARY=English_United States.utf8</span></span>
<span id="cb18-12"><a href="#cb18-12" tabindex="-1"></a><span class="co">#&gt; [4] LC_NUMERIC=C                          </span></span>
<span id="cb18-13"><a href="#cb18-13" tabindex="-1"></a><span class="co">#&gt; [5] LC_TIME=English_United States.utf8    </span></span>
<span id="cb18-14"><a href="#cb18-14" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb18-15"><a href="#cb18-15" tabindex="-1"></a><span class="co">#&gt; attached base packages:</span></span>
<span id="cb18-16"><a href="#cb18-16" tabindex="-1"></a><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span id="cb18-17"><a href="#cb18-17" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb18-18"><a href="#cb18-18" tabindex="-1"></a><span class="co">#&gt; other attached packages:</span></span>
<span id="cb18-19"><a href="#cb18-19" tabindex="-1"></a><span class="co">#&gt; [1] AGHmatrix_2.1.4 rmarkdown_2.23  knitr_1.43     </span></span>
<span id="cb18-20"><a href="#cb18-20" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb18-21"><a href="#cb18-21" tabindex="-1"></a><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span id="cb18-22"><a href="#cb18-22" tabindex="-1"></a><span class="co">#&gt;  [1] codetools_0.2-19  lattice_0.20-45   zoo_1.8-11        digest_0.6.31    </span></span>
<span id="cb18-23"><a href="#cb18-23" tabindex="-1"></a><span class="co">#&gt;  [5] grid_4.2.3        R6_2.5.1          jsonlite_1.8.7    evaluate_0.21    </span></span>
<span id="cb18-24"><a href="#cb18-24" tabindex="-1"></a><span class="co">#&gt;  [9] highr_0.10        cachem_1.0.7      rlang_1.1.0       cli_3.6.1        </span></span>
<span id="cb18-25"><a href="#cb18-25" tabindex="-1"></a><span class="co">#&gt; [13] rstudioapi_0.15.0 jquerylib_0.1.4   Matrix_1.6-0      bslib_0.5.0      </span></span>
<span id="cb18-26"><a href="#cb18-26" tabindex="-1"></a><span class="co">#&gt; [17] tools_4.2.3       xfun_0.39         yaml_2.3.7        fastmap_1.1.1    </span></span>
<span id="cb18-27"><a href="#cb18-27" tabindex="-1"></a><span class="co">#&gt; [21] compiler_4.2.3    htmltools_0.5.5   sass_0.4.7</span></span></code></pre></div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
