<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Kevin Wright" />


<title>Additional sources of agricultural data</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>







<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Additional sources of agricultural
data</h1>
<h4 class="author">Kevin Wright</h4>



<div id="other" class="section level1">
<h1>Other</h1>
<div id="rothamsted-library" class="section level2">
<h2>Rothamsted Library</h2>
<p>https://www.rothamsted.ac.uk/library-and-information-services</p>
<p>This has now been scanned and PDFs put into the GitHub repository for
the agridat package.</p>
<p>Box of uniformity trial data</p>
<pre><code>STATS17  WG Cochran
 
1. Uniformity trial data. 
2. Genstat data. Data received since publication of the catalogue.  1935-1943. 
3. Uniformity trial data.  1930-1936. 
4. Uniformity trials.  1936-1938. 
5. Uniformity trials.  R data.  1936-1937. 
6. O. V. S. Heath.  Cotton uniformity trial data.  1934-1935. 
7. Data.  Yields of grain per foot length.  1934. 
8. Catalogue of field uniformity trial data.  N. d. 
9. Demandt.  1931. One box </code></pre>
</div>
</div>
<div id="books" class="section level1">
<h1>Books</h1>
<div id="die-landwirtschaftlichen-versuchs-stations" class="section level3">
<h3>“Die Landwirtschaftlichen Versuchs-Stations”</h3>
<p>https://catalog.hathitrust.org/Record/000549685</p>
<p>Full view of research station reports 1859-1920. In German.</p>
</div>
<div id="d.-f.-andrews-and-a.-m.-herzberg-1985.-data." class="section level3">
<h3>D. F. Andrews and A. M. Herzberg (1985). “Data”.</h3>
<p>https://www2.stat.duke.edu/courses/Spring01/sta114/data/andrews.html</p>
<pre><code>Table  2.1: agridat::darwin.maize 
Table  5.1: agridat::broadbalk.wheat 
Table  6.1: agridat::mercer.wheat.uniformity 
Table  6.2: agridat::wiebe.wheat.uniformity 
Table 58.1: agridat::caribbean.maize</code></pre>
</div>
<div id="gemechu-application-of-spatial-mixed-model-in-agricultural-field-experiment" class="section level3">
<h3>Gemechu, “Application of Spatial Mixed Model in Agricultural Field
Experiment”</h3>
<p>Dibaba Bayisa Gemechu and Aweke, Girma (or maybe Girma Taye)</p>
<p>Master thesis. Department of Statistics, Addis Ababa University. One
dataset from wheat, RCB, with field coordinates.</p>
<p>Note: Forkman cites this author as “D. Bayisa”</p>
</div>
<div id="m.-n.-das-narayan-c.-giri-1987.-design-and-analysis-of-experiments." class="section level3">
<h3>M. N. Das &amp; Narayan C. Giri (1987). “Design and Analysis of
Experiments”.</h3>
<pre><code> 31 wool from 24 ewes, 6 cuttings
116 grass NPK factorial, 3 years, 36 obs
116 2^5 factorial, 1 rep, 32 obs
117 2^3 factorial, 3 rep
117 sugar beet 3^3 factorial, 2 rep, 54 obs
139 alfalfa 3x2^2 factorial
149 cabbage NPK split-plot, xy, 2 rep, 108 obs
150 soybean nitro-variety split-plot
193 wheat variety inc block, 9 block
201 rice variety balanced lattice, 80 obs
279 maize covariate, yield &amp; plant count, 4 rep, 32 obs</code></pre>
</div>
<div id="peter-diggle-patrick-heagerty-kung-yee-liang-scott-zeger.-analysis-of-longitudinal-data." class="section level3">
<h3>Peter Diggle, Patrick Heagerty, Kung-Yee Liang, Scott Zeger.
“Analysis of Longitudinal Data”.</h3>
<p>https://faculty.washington.edu/heagerty/Books/AnalysisLongitudinal/datasets.html</p>
<p>Pig weight data is found in <code>SemiPar::pig.weights</code></p>
<p>Sitka spruce data is found in: <code>geepack::spruce</code></p>
<p>Milk protein data is found in: <code>nlme::Milk</code>. A thorough
description of this data can be found in Molenberghs &amp; Kenward,
“Missing Data in Clinical Studies”, p. 377. Original source: A. P.
Verbyla and B. R. Cullis, Modelling in Repeated Measures
Experiments.</p>
</div>
<div id="federer-walt-1955.-experimental-design." class="section level3">
<h3>Federer, Walt (1955). “Experimental Design”.</h3>
<pre><code>192 3x3 factorial 
204 3x2 factorial 
236 2x2x2 factorial with confounding 
257 2x3x2 factorial with confounding 
276 split-plot with layout 
285 nested multi-loc (Also problems page 22) 
350 cubic lattice 
420 balanced inc block 
491 Latin square with covariate</code></pre>
</div>
<div id="finney-1972.-an-introduction-to-statistical-science-in-agriculture." class="section level3">
<h3>Finney 1972. “An Introduction to Statistical Science in
Agriculture”.</h3>
<p>Small, mostly simulated data.</p>
</div>
<div id="galwey-n.w.-2014.-introduction-to-mixed-modelling-2nd-ed." class="section level3">
<h3>Galwey, N.W. (2014). “Introduction to Mixed Modelling”, 2nd ed.</h3>
<p>https://www.wiley.com/en-us/Introduction+to+Mixed+Modelling%3A+Beyond+Regression+and+Analysis+of+Variance%2C+2nd+Edition-p-9781119945499</p>
<pre><code> 2  83 variety x nitro split-plot - agridat::yates.oats
 3 104 doubled-haploid barley
 3 135 wheat/rye competition, heritability
 5 190 chickpea flowering in families
 7 250 canola oil gxe, sowing date, rainfall, oil.  Si &amp; Walton 2004.
 7 284 pig growth, 4 diets
 7 285 sheep milk fat and lactose
 7 290 wheat anoxia root porosity, 9 gen
 7 291 wool fibers, 3 trt, 21 animals
 9 370 alphalpha design (row-column inc block for 2 reps) (not latinized row col)
10 434 hollamby wheat trial - agridat::gilmour.serpentine</code></pre>
</div>
<div id="grover-deepak-lajpat-rai.-experimental-designing-and-data-analysis-in-agriculture-and-biology." class="section level3">
<h3>Grover, Deepak &amp; Lajpat Rai. “Experimental Designing And Data
Analysis In Agriculture And Biology”.</h3>
<p>Agrotech Publishing Academy, 2010.
https://archive.org/details/expldesnanddatanalinagblg00023</p>
<pre><code>43 Percent insect survival in 12 rice varieties, 3 reps
50 CRD
57 RCBD
67 Latin Square
85 Sampling, 4 rep, 9 trt, 4 sub-samples agridat::grover.rcb.subsample
88 Split-plot, 3 rep, 2 measurements/plot, plant height (unusual subsample example)
97 Missing plot
105 Latin square with missing plot
115 2^2 factorial, 6 block
118 2^3 factorial, 3 block
120 Two factor asymmetrical, 5 rep
140 2^3 fractional factorial, 3 rep
160 Split-plot (planting date, variety), 3 rep
168 Strip-plot, 3 rep 
176 Milk yield with covariate
188 Multi-year nitrogen treatment
197 BIBD 13 varieties
205 Lattice 4 blocks, 3 reps, 16 trt
226 Augmented BIBD
236 Group-divisible
239 PBIB
241 Augmented group-divisible
245 Augmented PBIB
250 6x6 full diallel, 4 rep   agridat::grover.diallel</code></pre>
</div>
<div id="o.v.s.-heath-1970.-investigation-by-experiment." class="section level3">
<h3>O.V.S. Heath (1970). “Investigation by experiment”.</h3>
<p>https://archive.org/details/investigationbye0000heat</p>
<pre><code>23 uniformity trial of radish - agridat::heath.raddish.uniformity
50 uniformity trial of cabbage - agridat::heath.cabbage.uniformity</code></pre>
</div>
<div id="kwanchai-a.-gomez-gomez-1984.-statistical-procedures-for-agricultural-research." class="section level3">
<h3>Kwanchai A. Gomez &amp; Gomez (1984). “Statistical Procedures for
Agricultural Research”.</h3>
<p>Extensive collection of datasets from rice experiments. Many added to
agridat.</p>
</div>
<div id="cyril-h.-goulden-1939-methods-of-statistical-analysis." class="section level3">
<h3>Cyril H. Goulden (1939), “Methods of Statistical Analysis”.</h3>
<p>First edition:
https://archive.org/details/methodsofstatist031744mbp</p>
<pre><code> 18 Uniformity trial - agridat::goulden.barley.uniformity 
153 Split-split plot with factorial sub-plot treatment - agridat::goulden.splitsplit 
194 Incomplete block 
197 Inc block 
205 Latin square 
208 Inc block 
255 Covariates in feeding trial - agridat::crampton.pig</code></pre>
<p>Second edition:
<code>http://krishikosh.egranth.ac.in/handle/1/2034118</code>
(broken)</p>
<pre><code>216 Latin square - agridat::goulden.latin 
423 Control chart with egg weights - agridat::goulden.eggs</code></pre>
</div>
<div id="harry-love-1936.-applications-of-statistical-methods-to-agricultural-research." class="section level3">
<h3>Harry Love (1936). “Applications of Statistical Methods to
Agricultural Research”.</h3>
<pre><code>379 MET 4 year, 2 field, 5 block, 5 gen</code></pre>
</div>
<div id="kang-manjit-2003.-handbook-of-formulas-and-software-for-plant-geneticists-and-breeders" class="section level3">
<h3>Kang, Manjit (2003). “Handbook of Formulas and Software for Plant
Geneticists and Breeders”</h3>
</div>
<div id="kuehl-robert.-design-of-experiments-2nd-ed." class="section level3">
<h3>Kuehl, Robert. “Design of Experiments”, 2nd ed.</h3>
<pre><code>357 alfalfa quadruple lattice
358 alpha design
488 split-plot sorghum hybrid,density
516 alfalfa rcb, two-year
521 crossover design cattle feedstuff</code></pre>
</div>
<div id="erwin-leclerg-warren-leonard-andrew-clark-1962.-field-plot-technique" class="section level3">
<h3>Erwin LeClerg, Warren Leonard, Andrew Clark (1962). “Field Plot
Technique”</h3>
<p>https://archive.org/details/fieldplottechniq00leon</p>
<p>Many small datasets.</p>
<pre><code> 27 uniformity - agridat::goulden.barley.uniformity
213 split-plot
234 immer multi-environment
260 lattice pinto-bean
276 triple lattice cotton
280 lattice sugar beet
289 balanced lattice
336 repeated wheat</code></pre>
</div>
<div id="thomas-m-little-f.-jackson-hills-1978.-agricultural-experimentation." class="section level3">
<h3>Thomas M Little &amp; F. Jackson Hills (1978). “Agricultural
Experimentation”.</h3>
<pre><code> 79 Latin square 
 89 Split-plot 
103 Split-split 
117 Split-block - agridat::little.splitblock 
126 Repeated harvests.  In data-unused.
144 Non-IID errors
155 Square root transform
158 Germination, 3 reps, 24 treatments
261 Response surface, nitrogen, harvest
277 Count data</code></pre>
</div>
<div id="harald-martens-magni-martens.-multivariate-analysis-of-quality" class="section level3">
<h3>Harald Martens &amp; Magni Martens. “Multivariate Analysis of
Quality”</h3>
<p>https://www.wiley.com/legacy/wileychi/chemometrics/datasets.html</p>
<p>The ‘NIR’ data has NIR spectra measurements of wheat for the purpose
of understanding protein quality.</p>
</div>
<div id="roger-mead-robert-n.-curnow-anne-m.-hasted-2002.-statistical-methods-in-agriculture-and-experimental-biology-3rd-ed." class="section level3">
<h3>Roger Mead, Robert N. Curnow, Anne M. Hasted (2002). “Statistical
Methods in Agriculture and Experimental Biology”, 3rd ed.</h3>
<pre><code> 10 weekly milk yields
 24 carrot weight
 96 cabbage fertilizer
143 intercropping cowpea maize
177 honeybee repellent non-normal
251 cauliflower poisson - agridat::mead.cauliflower
273 rhubarb RCB covariate
296 onion density
316 lambs
341 germination
350 germination factorial - agridat::mead.germination
352 poppy
359 lamb loglinear - agridat::mead.lambs
375 rats
386 intercrop
390 intercrop cowpea maize - agridat::mead.cowpeamaize
404 apple characteristics (incomplete)</code></pre>
</div>
<div id="roger-mead-1988.-the-design-of-experiments" class="section level3">
<h3>Roger Mead (1988). “The Design of Experiments”</h3>
<p>https://books.google.com/books?id=CaFZPbCllrMC&amp;pg=PA323</p>
<pre><code>323 Turnip spacing data - agridat::mead.turnip</code></pre>
</div>
<div id="leonard-c.-onyiah-2008." class="section level3">
<h3>Leonard C. Onyiah (2008).</h3>
<p>“Design and Analysis of Experiments: Classical and Regression
Approaches with SAS”.
https://books.google.com/books?id=_P3LBQAAQBAJ&amp;pg=PA334</p>
<pre><code>334 Two examples of 5x5 Graeco-Latin squares in cassava and maize</code></pre>
</div>
<div id="bernard-ostle-1963.-statistics-in-research-2nd-ed." class="section level3">
<h3>Bernard Ostle (1963). “Statistics in Research”, 2nd ed.</h3>
<p>https://archive.org/details/secondeditionsta001000mbp</p>
<pre><code>455 2 factors, 1 covariate - agridat::woodman.pig
458 1 factor, 2 covariates - agridat::crampton.pig</code></pre>
</div>
<div id="v.-g.-panse-and-p.-v.-sukhatme-1957.-statistical-methods-for-agricultural-workers." class="section level3">
<h3>V. G. Panse and P. V. Sukhatme (1957). “Statistical Methods for
Agricultural Workers”.</h3>
<pre><code>  3 Length and number of grains per ear of wheat
138 Uniformity trial - agridat::panse.cotton.uniformity
154 RCB 8 blocks
167 two factorial, 6 rep trial
178 2^4 factorial, 8 blocks, partial confounding
192 3^3 factorial, 3 reps/9 blocks, partial confounding
200 split-plot, 6 rep
212 strip-plot, 6 rep
219 cotton variety trial, yield &amp; stand counts
256 8x8 simpple lattice, 4 reps
282 5 varieties at 6 locations
295 5 N levels at 5 locations
332 4 regions, 9-11 villages in each region, 3 fertilizer treatments
</code></pre>
<p>Note: The 1954 edition can be found at
https://archive.org/details/dli.scoerat.949statisticalmethodsforagriculturalworkers/page/138/mode/2up</p>
</div>
<div id="d.-d.-paterson-1939.-statistical-technique-in-agricultural-research." class="section level3">
<h3>D. D. Paterson (1939). “Statistical Technique in Agricultural
Research”.</h3>
<p>https://archive.org/details/statisticaltechn031729mbp</p>
<pre><code> 84 Distribution of purple/white starchy/sweet seeds from 11 ears
190 Sugar cane MET: 2 year, 5 block, 5 variety
199 Tea MET: 3 year, 2^2 factorial fertilizer
206 Grass: 4 rep, 2 gen, 4 cutting treatments
211 Cotton: 4 dates, 3 spacings, 3 irrigation, 2 nitro - agridat::gregory.cotton</code></pre>
</div>
<div id="roger-petersen-agricultural-field-experiments" class="section level3">
<h3>Roger Petersen, “Agricultural Field Experiments”</h3>
<pre><code>8 Uniformity trial 18 * 6 plots
56 RCB 4 rep, 5 trt
71 Latin square 5x5
86 Factorial 4x2, 3 rep
97 Factorial 2x3x2, 3 rep
125 Fertilizer trial, 3 rep, 5 levels
136 Split plot variety x planting date, 3 rep
148 Strip plot 2 potash x 3 potassium, 3 rep
170 Augmented breeding trial with 3 checks, 6 inc blocks
174 Inc Block
182 Lattice 5x5, 2 rep
192 GxE 10 gen, 12 env. Stability analysis.
208 Factorial 2x3 at 8 locs, homogeneous variance, early lentils
217 GxE 8 gen, 5 loc, heterogeneous variance
232 Factorial 2x3 at 8 locs, late lentils (see also page 208)
249 On-farm trial, 24 entries, 3 rep RCB
257 Demonstration trials, 5 locs
272 Covariance example, RCB 6 rep, 4rt
278 Multi-year 2x2 factorial, 4 rep
309 Pasture trial
323 On-farm trial, 2 variety 8 loc
327 On-farm trial 6 trt, 5 loc
334 On-farm trial 4 trt, 6 loc
343 On-farm trial 2x3 factorial, 3 loc
351 Feeding trial, 2 trt, 2 periods
357 Intercrop, 2 crops
372 Intercrop, 2 crop, 4 mixtures, 4 rep. agridat::petersen.sorghum.cowpea</code></pre>
</div>
<div id="richard-plant-spatial-data-analysis-in-ecology-and-agriculture-using-r" class="section level3">
<h3>Richard Plant, “Spatial Data Analysis in Ecology and Agriculture
using R”</h3>
<p>https://psfaculty.plantsciences.ucdavis.edu/plant/</p>
</div>
<div id="arthur-asquith-rayner-1969.-a-first-course-in-biometry-for-agriculture-students." class="section level3">
<h3>Arthur Asquith Rayner (1969). “A First Course In Biometry For
Agriculture Students”.</h3>
<pre><code>19 456 2x2x4 Factorial, 2 rep
19 466 2x4 factorial, layout, plot size, kale (from Rothamsted)
19 466 3x5 factorial, 3 rep, potato
20 494 3x4 Split-plot with layout
21 505 2x2x2 Factorial, 5 rep
21 515 2x2x2x2 Factorial, 3 rep, with layout. (Evaluated, rejected as too variable)
22 537 2x2x2 factorial, 6 rep, potato
22 537 2x2x2x2 factorial, 2 rep, wheat, layout</code></pre>
</div>
<div id="f.s.f.-shaw-1936.-a-handbook-of-statistics-for-use-in-plant-breeding-and-agricultural-problems" class="section level3">
<h3>F.S.F. Shaw (1936). “A Handbook of Statistics For Use in Plant
Breeding and Agricultural Problems”</h3>
<p>https://archive.org/details/in.ernet.dli.2015.176662</p>
<pre><code>5 Length of ear head and number of grains per ear, 400 ears.
95 variety RCB, 5 gen, 25 rep, diagonal layout
107 Latin square, 8 entries.
117 Factorial: 8 blocks, 3 varieties, 5 treatments, 2 infections
126 Multi-environment trial, 3 year, 13 varieties, 2 loc, 5 blocks  agridat::shaw.oats
</code></pre>
</div>
<div id="g.-w.-snedecor-w.-g.-cochran.-statistical-methods." class="section level3">
<h3>G. W. Snedecor &amp; W. G. Cochran. “Statistical Methods”.</h3>
<pre><code>168 regression
352 3x3 factorial, 4 blocks
359 2x2x2 factorial, 8 blocks, daily pig gain
362 2x3x4 factorial, 2 blocks, daily pig gain
371 3x4 split-plot, 3 var, 4 date, 6 blocks
374 2x3x3 split-split-plot, irrig, stand, fert, block
378 4x4 split-plot, 4 block, 4 year, 4 cuttings asparagus
384 regression with 2 predictors
428 covariates, 6 varieties, 4 blocks, yield vs stand
440 pig gain vs initial weight, 4 treatments, 40 pigs
454 protein vs yield for wheat, 91 plots, quadratic regression
</code></pre>
</div>
<div id="robert-g.-d.-steel-james-hiram-torrie.-principles-and-procedures-of-statistics-2nd-ed." class="section level3">
<h3>Robert G. D. Steel &amp; James Hiram Torrie. “Principles and
Procedures of Statistics”, 2nd ed.</h3>
<pre><code>154 Mint plant growth, 2-way + pot + plant
244 Trivariate data
319 Regression with three predictors
384 Split-plot yield
387 Split-plot row spacing
400 Soybean 3 loc
423 Pig weight gain
429 Guinea pig weight gain
434 Soybean lodging</code></pre>
</div>
<div id="oliver-schabenberger-and-francis-j.-pierce.-contemporary-statistical-models-for-the-plant-and-soil-sciences." class="section level3">
<h3>Oliver Schabenberger and Francis J. Pierce. “Contemporary
Statistical Models for the Plant and Soil Sciences”.</h3>
<p>Many datasets. Some added to agridat.</p>
</div>
<div id="s.-j.-welham-et-al.-2015.-statistical-methods-in-biology." class="section level3">
<h3>S. J. Welham et al. (2015). “Statistical Methods In Biology”.</h3>
<p>The online-supplements contain many small datasets for the examples
and exercises.</p>
</div>
<div id="pesticides-in-the-nations-streams-and-ground-water-1992-2001" class="section level3">
<h3>Pesticides in the Nation’s Streams and Ground Water, 1992-2001</h3>
<p>Extensive data for detection of pesticides in water samples. See
Appendix 5 and Appendix 6 of the supporting info.
https://water.usgs.gov/nawqa/pnsp/pubs/circ1291/supporting_info.php</p>
</div>
</div>
<div id="data-repositories" class="section level1">
<h1>Data Repositories</h1>
<div id="ag-data-commons" class="section level3">
<h3>Ag Data Commons</h3>
<p>https://data.nal.usda.gov/about-ag-data-commons
https://data.nal.usda.gov/search/type/dataset</p>
</div>
<div id="cyverse-data-commons" class="section level3">
<h3>CyVerse Data Commons</h3>
<p>https://datacommons.cyverse.org/
https://datacommons.cyverse.org/browse/iplant/home/shared/commons_repo/curated</p>
</div>
<div id="datadryad" class="section level3">
<h3>DataDryad</h3>
</div>
<div id="harvard-dataverse" class="section level3">
<h3>Harvard Dataverse</h3>
<p>https://dataverse.harvard.edu/</p>
<p>IRRI Rice Research includes plot-level data for long term rice
experiments. https://dataverse.harvard.edu/dataverse/RiceResearch</p>
</div>
<div id="kellogg-biological-station-long-term-research" class="section level3">
<h3>Kellogg Biological Station Long-Term Research</h3>
<p>KBS037:Precision Agriculture Yield Monitoring in Row Crop Agriculture
https://lter.kbs.msu.edu/datasets/40
https://doi.org/10.6073/pasta/423c07d6ea3317c545beabb4b8e502c8 Yield
monitor data across several years and crops. Un-friendly license.</p>
</div>
<div id="nature-scientific-data" class="section level3">
<h3>Nature Scientific Data</h3>
<p>https://www.nature.com/sdata/</p>
</div>
<div id="open-data-journal-for-agricultural-research" class="section level3">
<h3>Open Data Journal for Agricultural Research</h3>
<p>https://library.wur.nl/ojs/index.php/odjar/</p>
</div>
<div id="plant-genomics-and-phenomics-research-data-repository" class="section level3">
<h3>Plant Genomics and Phenomics Research Data Repository</h3>
</div>
<div id="wolfram-data-repository" class="section level3">
<h3>Wolfram Data Repository</h3>
<p>https://datarepository.wolframcloud.com/</p>
</div>
</div>
<div id="journals---bulletins" class="section level1">
<h1>Journals - Bulletins</h1>
<div id="iowa-state-agricultural-research-bulletins" class="section level3">
<h3>Iowa State Agricultural Research Bulletins</h3>
<p>https://lib.dr.iastate.edu/ag_researchbulletins/</p>
<pre><code>Vol 26/ 281. Cox: Analysis of Lattice and Triple Lattice.
Page 11: Lattice, 81 hybs, 4 reps 
Page 24: Triple lattice, 81 hybs, 6 reps

Vol 29/347. Homeyer. Punched Card and Calculating Machine Methods for Analyzing 
  Lattice Experiments Including Lattice Squares and the Cubic Lattice.
Page 37: Triple lattice (9 blocks * 9 hybrids) with 6 reps.
Page 60: Simple lattice, 8 blocks * 8 hybrids, 4 reps.
Page 76: Balanced lattice, 25 hybrids 
Page 87: Lattice square with (k+1)/2 reps, 121 hybrids, 6 rep 
Page 109: Lattice square with k+1 reps, 7 blocks * 7 hyb, 8 reps 
Page 126: Cubic lattice, 16 blocks * 4 plots = 64 varieties, 9 reps, cotton

Vol 32/396. Wassom. Bromegrass Uniformity Trial: agridat::wassom.bromegrass.uniformity

Vol 33/424. Heady. Crop Response Surfaces and Economic Optima in 
  Fertilizer - agridat::heady.fertilizer

Vol 34/358. Schwab. Research on Irrigation of Corn and Soybeans At Conesville.
Page 257. 2 year, 2 loc, 4 rep, 2 nitro. Stand &amp; yield.
  Nice graph of soil moisture deficit (fig 9)

Vol. 34/463. Doll. Fertilizer Production Functions for Corn and Oats.
Table 1, 1954 Clarion Loam.  N,P,K.
Table 14, 1955 McPaul Silt Loam.  N,P.
Table 25, 1955 corn.  K,P,N.
Table 31, 1956 oats, K,P,N.  Trends difficult to establish.

Vol 34/472. Pesek. Production Surfaces and Economic Optima For Corn Yields.
  Same data published in SSA journal?

Vol 34/488. Walker. Application of Game Theory Models to Decisions.

Vol 35/494. North Central Regional Potassium Studies with Alfalfa.
Page 176. Two years, several locs per state, multiple states, multiple fertilizer 
  levels, multiple cuttings. Soil test attributes.
Page 183. Yield and %K.

Vol 35/503. North Central Regional Potassium Studies with Corn.</code></pre>
</div>
</div>
<div id="papers" class="section level1">
<h1>Papers</h1>
<div id="bakare-et-al" class="section level3">
<h3>Bakare et al</h3>
<p>Exploring genotype by environment interaction on cassava yield and
yield related traits using classical statistical methods
https://doi.org/10.1371/journal.pone.0268189</p>
<p>36 gen, 20 env, 3 rep. Analysis and data here:
https://github.com/mab658/classical_analysis_GxE</p>
</div>
<div id="chaves-2023-et-al" class="section level3">
<h3>Chaves 2023 et al</h3>
<p>Analysis of multi-harvest data through mixed models: an application
in Theobroma grandiflorum breeding
https://acsess.onlinelibrary.wiley.com/doi/epdf/10.1002/csc2.20995 Nice.
Complete data and R code. They found FA3 best for genetic covariances,
AR1H best for residual structure. Used FAST and OP (by Cullis) for
selection.</p>
</div>
<div id="cleveland-m.a.-and-john-m.-hickey-selma-forni-2012." class="section level3">
<h3>Cleveland, M.A. and John M. Hickey, Selma Forni (2012).</h3>
<p>A Common Dataset for Genomic Analysis of Livestock Populations. G3,
2, 429-435. https://doi.org/10.1534/g3.111.001453</p>
<p>The supplemental information for this paper contains data for 3534
pigs with high-density genotypes (50000 SNPs), and a pedigree including
parents and grandparents of the animals.</p>
</div>
<div id="coelho-2021-et-al" class="section level3">
<h3>Coelho 2021 et al</h3>
<p>Accounting for spatial trends in multi-environment diallel analysis
in maize breeding https://doi.org/10.1371/journal.pone.0258473</p>
<p>78 hybrids in a diallel, 4 environments, 3 reps. Compared spatial and
non-spatial analyses.</p>
</div>
<div id="daillant-spinnler-1996.-relationships-between-perceived-sensory-properties-and-major-preference-directions-of-12-variaties-of-apples-from-the-southern-hemisphere." class="section level3">
<h3>Daillant-Spinnler (1996). Relationships between perceived sensory
properties and major preference directions of 12 variaties of apples
from the southern hemisphere.</h3>
<p>Food Quality and Preference, 7(2), 113-126.
https://doi.org/10.1016/0950-3293(95)00043-7</p>
<p>The data are in <code>ClustVarLV::apples_sh$pref</code> and
<code>ClustVarLV::apples_sh$senso</code> 12 apple varieties, 43 traits,
60 consumers</p>
</div>
<div id="gregory-crowther-lambert-1932.-the-interrelation-of-factors-controlling-the-production-of-cotton-under-irrigation-in-the-sudan." class="section level3">
<h3>Gregory, Crowther &amp; Lambert (1932). The interrelation of factors
controlling the production of cotton under irrigation in the Sudan.</h3>
<p>Jour Agric Sci, 22, p. 617.</p>
</div>
<div id="hedrick-1920." class="section level3">
<h3>Hedrick (1920).</h3>
<p>Twenty years of fertilizers in an apple orchard.
https://books.google.com/books?hl=en&amp;lr=&amp;id=SqlJAAAAMAAJ&amp;oi=fnd&amp;pg=PA446</p>
<p>The authors found no significant differences between fertilizer
treatments.</p>
</div>
<div id="meehan-gratton-2016." class="section level3">
<h3>Meehan &amp; Gratton (2016).</h3>
<p>A Landscape View of Agricultural Insecticide Use across the
Conterminous US from 1997 through 2012. PLOS ONE,
https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0166724</p>
<p>Supplemental material contains county-level data for each of 4 years.
Complete R-INLA code for analysis.</p>
</div>
<div id="monteverde-et-al" class="section level3">
<h3>Monteverde et al</h3>
<p>Integrating Molecular Markers and Environmental Covariates To
Interpret Genotype by Environment Interaction in Rice (Oryza sativa L.)
Grown in Subtropical Areas https://doi.org/10.1534/g3.119.400064
https://gsajournals.figshare.com/articles/dataset/Supplemental_Material_for_Monteverde_et_al_2019/7685636</p>
<p>Supplemental information contains phenotypic data and markers and
environmental covariates for PLS analysis.</p>
</div>
<div id="kenward-michael-g.-1987." class="section level3">
<h3>Kenward, Michael G. (1987).</h3>
<p>A Method for Comparing Profiles of Repeated Measurements. Applied
Statistics, 36, 296-308.</p>
<p>An ante-dependence model is fit to repeated measures of cattle
weight.</p>
</div>
<div id="klumper-qaim-2015." class="section level3">
<h3>Klumper &amp; Qaim (2015).</h3>
<p>A Meta-Analysis of the Impacts of Genetically Modified Crops.
https://doi.org/10.1371/journal.pone.0111629</p>
<p>Nice meta-analysis dataset. Published data only include differences,
not standard-errors. See the comments on PLOS article for some
peculiarities in the data.</p>
</div>
<div id="lado-b.-et-al.-2013." class="section level3">
<h3>Lado, B. et al. (2013).</h3>
<p>“Increased Genomic Prediction Accuracy in Wheat Breeding Through
Spatial Adjustment of Field Trial Data”. G3, 3, 2105-2114.
https://doi.org/10.1534/g3.113.007807</p>
<p>Has a large haplotype dataset (83 MB) and two-year phenotype data
with multiple traits.</p>
</div>
<div id="oakey-cullis-thompson-2016" class="section level3">
<h3>Oakey, Cullis, Thompson 2016</h3>
<p>Genomic Selection in Multi-environment Crop Trials
https://www.g3journal.org/content/6/5/1313
http://www.g3journal.org/content/6/5/1313/suppl/DC1 648 genotypes
planted in pots yr 1, 856 lines yr 2, 639 common to both years. 7864 SNP
markerks</p>
</div>
<div id="peixoto-marco-antonio-et-al-2020" class="section level3">
<h3>Peixoto, Marco Antonio et al (2020)</h3>
<p>Random regression for modeling yield genetic trajectories in Jatropha
curcas breeding.
https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0244021</p>
<p>Repeated measurements over six years. Data in supplemental Word
doc.</p>
</div>
<div id="perez-valencia-2022." class="section level3">
<h3>Perez-Valencia (2022).</h3>
<p>A two‑stage approach for the spatio‑temporal analysis of
high‑throughput phenotyping data.
https://doi.org/10.1038/s41598-022-06935-9</p>
<p>Time-series data for individual plots in a field of many
genotypes.</p>
</div>
<div id="roger-w.-hexem-earl-o.heady-metin-caglar-1974" class="section level3">
<h3>Roger W. Hexem, Earl O.Heady, Metin Caglar (1974)</h3>
<p>A compendium of experimental data for corn, wheat, cotton and sugar
beets grown at selected sites in the western United States and
alternative production functions fitted to these data. Technical report:
Center for Agricultural and Rural Development, Iowa State University.
https://babel.hathitrust.org/cgi/pt?id=wu.89031116783;view=1up;seq=3</p>
<p>The technical report provides data from experiments on corn, wheat,
cotton &amp; sugar beets, each crop tested at several locations over two
years, with a factorial structure on irrigation and nitrogen treatments,
with replications. Three polynomial functions were fit to the data for
each location (quadratic, square root, three-halves).</p>
</div>
<div id="snedecor-george-and-e.-s.-haber-1946." class="section level3">
<h3>Snedecor, George and E. S. Haber (1946).</h3>
<p>Statistical Methods For an Incomplete Experiment on a Perennial Crop.
Biometrics Bulletin, 2, 61-67. https://www.jstor.org/stable/3001959</p>
<p>Harvest of asparagus over 10 years, three cutting dates per year, 6
blocks.</p>
</div>
<div id="tanaka-takashi-x.-t." class="section level3">
<h3>Tanaka, Takashi X. T.</h3>
<p>Assessment of design and analysis frameworks for on-farm
experimentation through a simulation study of wheat yield in Japan.
https://github.com/takashit754/geostat</p>
<p>Yield-monitor data for 3 fields.</p>
</div>
<div id="technow-frank-et-al.-2014." class="section level3">
<h3>Technow, Frank, et al. (2014).</h3>
<p>Genome Properties and Prospects of Genomic Prediction of Hybrid
Performance in a Breeding Program of Maize. August 1, 2014 vol. 197 no.
4 1343-1355. https://doi.org/10.1534/genetics.114.165860</p>
<p>Genotype and phenotype data appears in the sommer package.</p>
</div>
<div id="tian-ting-2015." class="section level3">
<h3>Tian, Ting (2015).</h3>
<p>Application of Multiple Imputation for Missing Values in Three-Way
Three-Mode Multi-Environment Trial Data.
https://doi.org/10.1371/journal.pone.0144370</p>
<p>Uses <code>agridat::australia.soybean</code> data and one other real
dataset with 4 traits that are not identified. All data and code
available.</p>
</div>
<div id="randall-j.-wisser-et-al.-2011." class="section level3">
<h3>Randall J. Wisser et al. (2011).</h3>
<p>Multivariate analysis of maize disease resistances suggests a
pleiotropic genetic basis and implicates a GST gene. PNAS.
https://doi.org/10.1073/pnas.1011739108</p>
<p>The supplement contains genotype data, but no phenotype data.</p>
</div>
<div id="rife-et-al.-2018" class="section level3">
<h3>Rife et al. (2018)</h3>
<p>Genomic analysis and prediction within a US public collaborative
winter wheat regional testing nursery.
https://doi.org/10.5061/dryad.q968v83</p>
<p>Large phenotypic dataset with 691 wheat lines, 33 years, 670
environments, 3-4 reps, 120000 datapoints. No genotypic data is
included.</p>
</div>
<div id="schmitz-carley-et-al-2018" class="section level3">
<h3>Schmitz Carley et al (2018)</h3>
<p>Genetic Covariance of Environments in the Potato National Chip
Processing Trial
https://dl.sciencesocieties.org/publications/cs/articles/59/1/107</p>
<p>Supp 2 contains genomic data, but there is no easy way to find the
phenotypic data.</p>
</div>
<div id="van-der-voet-et-al.-2017." class="section level3">
<h3>van der Voet et al. (2017).</h3>
<p>Equivalence testing using existing reference data: An example with
genetically modified and conventional crops in animal feeding studies.
https://doi.org/10.1016/j.fct.2017.09.044</p>
<p>The full datasets for the GRACE studies A-E are available here:
https://www.cadima.info/index.php/area/publicAnimalFeedingTrials CC
license.</p>
</div>
<div id="volpato-et-al-2024" class="section level3">
<h3>Volpato et al (2024)</h3>
<p>A retrospective analysis of historical data of multi-environment
trials for dry bean ( Phaseolus vulgaris L.) in Michigan.
https://github.com/msudrybeanbreeding/DryBean_MultiEnvTrials Full
dataset and R code.</p>
</div>
<div id="xavier-alencar-et-al.." class="section level3">
<h3>Xavier, Alencar et al..</h3>
<p>Genome-Wide Analysis of Grain Yield Stability and Environmental
Interactions in a Multiparental Soybean Population.
https://doi.org/10.1534/g3.117.300300</p>
<p>Data are in the SoyNAM and NAM packages.</p>
</div>
<div id="yan-weikei-2002." class="section level3">
<h3>Yan, Weikei (2002).</h3>
<p>Singular value partitioning in biplots. Agron Journal.</p>
<p>Winter wheat, 31 gen in 8 loc. This data is different from Yan’s
earlier papers. Unfortunately, the data given in the paper are missing
two rows.</p>
</div>
</div>
<div id="r-packages-on-cran-github-etc." class="section level1">
<h1>R packages on CRAN, Github, etc.</h1>
<p>See also: https://cran.r-project.org/web/views/Agriculture.html</p>
<div id="agml" class="section level3">
<h3>AgML</h3>
<p>https://github.com/Project-AgML/AgML</p>
<p>Datasets for agricultural machine learning such as image
classification, semantic segmentation, object detection, etc.</p>
</div>
<div id="agricensdata" class="section level3">
<h3>agriCensData</h3>
<p>https://github.com/OnofriAndreaPG/agriCensData</p>
<p>Three datasets with censored observations for the paper “Analyzing
interval-censored data in agricultural research: A review with examples
and software tips”.</p>
</div>
<div id="agritutorial" class="section level3">
<h3>agriTutorial</h3>
<p>https://myaseen208.github.io/agriTutorial/</p>
<p>Five datasets used to illustrate analyses.</p>
</div>
<div id="agricolae" class="section level3">
<h3>agricolae</h3>
<p>Has assorted data and functions for analysis of agricultural
data.</p>
</div>
<div id="agrobiodata" class="section level3">
<h3>agroBioData</h3>
<p>https://github.com/OnofriAndreaPG/agroBioData</p>
<p>Datasets for agriculture and applied biology. Referenced by this
blog: https://www.statforbiology.com/</p>
</div>
<div id="aml---adaptive-mixed-lasso" class="section level3">
<h3>aml - Adaptive Mixed LASSO</h3>
<p>Data <code>aml::wheat</code> genetic and phenotypic data for wheat.
Modest size.</p>
</div>
<div id="bglr---bayesian-generalized-linear-regression." class="section level3">
<h3>BGLR - Bayesian Generalized Linear Regression.</h3>
<p>Has an A matrix (but no pedigree) for 499 genotypes at 4
locations.</p>
</div>
<div id="blr---bayesian-linear-regression." class="section level3">
<h3>BLR - Bayesian Linear Regression.</h3>
<p>Has an A matrix (but no pedigree) for 499 genotypes at 4
locations.</p>
</div>
<div id="bsagri" class="section level3">
<h3>BSagri</h3>
<p>Safety assessment in agriculture trials</p>
</div>
<div id="clustvarlv" class="section level3">
<h3>ClustVarLV</h3>
<p>Data <code>apples_sh</code> sensory attributes and preference scores
for 12 apple varieties.</p>
</div>
<div id="cropcc---climate-change-on-crops" class="section level3">
<h3>cropcc - Climate change on crops</h3>
<p>https://r-forge.r-project.org/projects/cropcc/</p>
</div>
<div id="drc---dose-response-curves" class="section level3">
<h3>drc - Dose response curves</h3>
<p>Has nice herbicide dose response curves and germination data for
mungbean, rice, wheat.</p>
</div>
<div id="epiphy" class="section level3">
<h3>epiphy</h3>
<p>https://github.com/chgigot/epiphy</p>
<p>Contains 10 historical datasets for plant disease epidemics.</p>
</div>
<div id="fw---finlay-wilkinson-regression" class="section level3">
<h3>FW - Finlay-Wilkinson regression</h3>
<p>https://github.com/lian0090/FW/</p>
<p>Has phenotype data and marker data for 599 wheat lines in 4
environments.</p>
</div>
<div id="ggenealogy" class="section level3">
<h3>ggenealogy</h3>
<p>https://doi.org/10.18637/jss.v089.i13</p>
<p>Data <code>sbGeneal</code> contains a soybean pedigree with 230
varieties.</p>
</div>
<div id="grbase" class="section level3">
<h3>gRbase</h3>
<p>Data <code>gRbase::carcass</code>: thickness of meat and fat on
slaughter pigs</p>
</div>
<div id="lmdiallel" class="section level3">
<h3>lmDiallel</h3>
<p>https://github.com/OnofriAndreaPG/lmDiallel/tree/master/data</p>
</div>
<div id="lmtest" class="section level3">
<h3>lmtest</h3>
<p>Data <code>lmtest::ChickEgg</code> time series of annual chicken and
egg production in the United States 1930-1983.</p>
</div>
<div id="nada" class="section level3">
<h3>NADA</h3>
<p>Data <code>Atra</code> and <code>Recon</code> contain measurements of
Atrazine in water samples.</p>
</div>
<div id="nlraa" class="section level3">
<h3>nlraa</h3>
<p>Miguez. Non-linear models in agriculture. <code>nlraa::sm</code> =
<code>agridat::miguez.biomass</code> Vignettes and functions for working
with (non)linear mixed models</p>
</div>
<div id="nlme" class="section level3">
<h3>nlme</h3>
<p><code>nlme::Orange</code>: Growth of orange trees</p>
<p><code>nlme::Soybean</code>: Growth of soybean plants. From the book
“Nonlinear Models for Repeated Measurement Data”.</p>
</div>
<div id="ofpe---on-farm-precision-experiments" class="section level3">
<h3>OFPE - On-Farm Precision Experiments</h3>
<p>https://paulhegedus.github.io/OFPE-Website/
https://github.com/paulhegedus/OFPEDATA/</p>
</div>
<div id="onfant.dataset" class="section level3">
<h3>onfant.dataset</h3>
<p>https://github.com/AnabelleLaurent/onfant.dataset</p>
</div>
<div id="pbkrtest" class="section level3">
<h3>pbkrtest</h3>
<p><code>pbkrtest::beets</code> Yield and percent sugar in split-plot
experiment.</p>
</div>
<div id="plantbreeding" class="section level3">
<h3>plantbreeding</h3>
<p>https://r-forge.r-project.org/projects/plantbreeding/</p>
<pre><code>Data: fulldial
Data: linetester
Data: peanut - same as agridat::kang.peanut</code></pre>
</div>
<div id="sdaa---survey-data-and-analysis" class="section level3">
<h3>SDaA - Survey Data and Analysis</h3>
<p>This package has county-level data from the United States Census of
Agriculture, along with a vignette to illustrate survey sampling
analyses.</p>
</div>
<div id="semipar" class="section level3">
<h3>SemiPar</h3>
<p>Data: <code>SemiPar::onions</code> is same as
agridat::ratkowski.onions</p>
</div>
<div id="soildb" class="section level3">
<h3>soilDB</h3>
<p>https://ncss-tech.github.io/AQP/soilDB/soilDB-Intro.html Soil
database interface.</p>
</div>
<div id="sommer---solving-mixed-model-equations-in-r" class="section level3">
<h3>sommer - Solving mixed model equations in R</h3>
<p>Data: h2. Modest-sized GxE experiment in potato Data: cornHybrid.
Yield/PLTHT for 100 hybrids from 20 inbred * 20 inbred, 4 locs.
Phenotype and relationship matrix.</p>
<p>Data:</p>
<pre><code>data(DT_wheat) #  CIMMYT wheat data
DT_wheat # 599 varieties, yield in 4 envts
GT_wheat # 599 varieties, 1279 markers coded -1,1</code></pre>
<p>Data: RICE</p>
<p>Data: FDdata taken from agridat::bond.diallel</p>
<p>Data:</p>
<pre><code>data(DT_technow) # From http://www.genetics.org/content/197/4/1343.supplemental
DT &lt;- DT_technow  # 1254 hybs, parents, GY=yield, GM=moisture
Md &lt;- Md_technow  # 123 dent parents, 35478 markers
Mf &lt;- Mf_technow  # 86 flint parents, 37478 markers
Ad &lt;- Ad_technow  # 123 x 123 A matrix 
Af &lt;- Af_technow  # 86 x 85 A matrix</code></pre>
</div>
<div id="soynam---soybean-nested-association-mapping" class="section level3">
<h3>SoyNAM - Soybean nested association mapping</h3>
<p>Dataset with phenotype data 3 yr, 9 locations, 18 environments, 60
thousand observations for height, maturity, lodging, moisture, protein,
oil, fiber, seed size. There are 5000+ strains, 40 families.</p>
<p>Data formatted for the analysis of the NAM package is available with
the following command: <code>SoyNAM::ENV()</code>.</p>
</div>
<div id="soyurt" class="section level3">
<h3>SoyURT</h3>
<p>https://github.com/mdkrause/SoyURT</p>
<p>Large historical data of yield trials from Uniform Soybean Tests
Northern States. Years 1989-2019, 63 locations, 4257 genotypes. The
package also contains soils and weather data for the trial
locations.</p>
<p>Note: The USDA published papers with results from: National Cotton
Variety Tests, Uniform Soybean Tests Northern States, and Uniform
Soybean Tests Southern States here:
https://www.ars.usda.gov/southeast-area/stoneville-ms/crop-genetics-research/docs/</p>
</div>
<div id="spdep" class="section level3">
<h3>spdep</h3>
<p>Has a vignette ‘The Problem of Spatial Autocorrelation: forty years
on’ that examines agriculture in Irish counties. See also the data
<code>ade4::irishdata</code>.</p>
</div>
<div id="spurs" class="section level3">
<h3>spuRs</h3>
<p>Data: <code>spuRs::trees</code> has data for 107 trees that were cut
into cross sections with the volume calculated at roughly 10-year
increments. This is a subset of the much-larger original data from
Guttenberg: https://archive.org/stream/wachstumundertra00gutt</p>
</div>
<div id="statforbiology" class="section level3">
<h3>StatForBiology</h3>
<p>https://www.statforbiology.com/</p>
<p>Blog posts with example analyses.</p>
</div>
<div id="biometris---statgengxe" class="section level3">
<h3>Biometris - statgenGxE</h3>
<p>https://github.com/Biometris/statgenGxE
https://biometris.github.io/statgenGxE/</p>
<p>AMMI, FW, GGE stability analyses.</p>
</div>
<div id="biometris---statgengwas" class="section level3">
<h3>Biometris - statgenGWAS</h3>
<p>https://github.com/Biometris/statgenGWAS/
https://CRAN.R-project.org/package=statgenGWAS</p>
<p>This is a very nice package with full GxE data and marker data with
41722 loci on 246 lines.</p>
<p>256 hybrids, 29 envts across 2 years, multi-trait (yield, silking,
pltht, earht, etc). Includes a worked example with data from:
https://data.inra.fr/dataset.xhtml?persistentId=doi:10.15454/IASSTN And
publication: Millet 2016, Genome-Wide Analysis of Yield in Europe:
Allelic Effects Vary with Drought and Heat Scenarios,
https://academic.oup.com/plphys/article/172/2/749/6115953</p>
</div>
<div id="biometris---statgensta" class="section level3">
<h3>Biometris - statgenSTA</h3>
<p>https://github.com/Biometris/statgenSTA/
https://CRAN.R-project.org/package=statgenSTA</p>
<p>Analysis of phenotypic data from field experiments using SpATS, lme4,
or asreml.</p>
</div>
<div id="st4gi---stat-for-genetic-improvement" class="section level3">
<h3>st4gi - Stat for genetic improvement</h3>
<p>https://github.com/reyzaguirre/st4gi</p>
</div>
</div>
<div id="web-sites" class="section level1">
<h1>Web sites</h1>
<div id="ars-oat-trials" class="section level3">
<h3>ARS oat trials</h3>
<p>https://www.ars.usda.gov/Main/docs.htm?docid=8419&amp;page=4</p>
</div>
<div id="cimmyt-research-data" class="section level3">
<h3>CIMMYT Research Data</h3>
<p>https://data.cimmyt.org/dataverse/cimmytdatadvn</p>
</div>
<div id="gardian-platform-for-big-data-in-agriculture" class="section level3">
<h3>Gardian Platform for Big Data in Agriculture</h3>
</div>
<div id="grain-genes" class="section level3">
<h3>Grain genes</h3>
<ol style="list-style-type: decimal">
<li><p>https://wheat.pw.usda.gov/ggpages/HxT/ The Harrington x TR306
Barley Mapping Population. The genotype and phenotype data comes from
Mapmaker, but seems to be in a slightly non-standard format; 145 DH
lines, 217 markers, 25 env, 1 rep.</p></li>
<li><p>https://wheat.pw.usda.gov/ggpages/SxM/ . This data is
agridat::steptoe.morex.</p></li>
</ol>
</div>
<div id="glten---a-network-of-long-term-trials-around-the-world" class="section level3">
<h3>GLTEN - A network of Long-Term trials around the world</h3>
<p>https://glten.org/</p>
</div>
<div id="ideals" class="section level3">
<h3>Ideals</h3>
<p>https://www.ideals.illinois.edu/handle/2142/3528 Data File : Raw data
from each ear analyzed each year of the Illinois long-term selection
experiment for oil and protein in corn (1896-2004)</p>
</div>
<div id="international-potato-center" class="section level3">
<h3>International Potato Center</h3>
<p>https://data.cipotato.org/dataverse.xhtml</p>
</div>
<div id="ilri-international-livestock-research-institute" class="section level3">
<h3>ILRI International Livestock Research Institute</h3>
<p>Case study 4 is a nice diallel example with sheep data. Available as
agridat::ilri.sheep</p>
</div>
<div id="irri-biometrics-and-breeding-informatics" class="section level3">
<h3>IRRI Biometrics and Breeding Informatics</h3>
<p>http://bbi.irri.org/products</p>
<p>STAR, PBTools, CropStat. The STAR user guide has well-documented data
(even using 2 from agridat), but the PBTools user guide does not
document the data.</p>
</div>
<div id="miappe-minimum-information-about-plant-phenotyping-experiments" class="section level3">
<h3>MIAPPE Minimum Information About Plant Phenotyping Experiments</h3>
<p>https://www.miappe.org/</p>
<p>Very limited data.</p>
</div>
<div id="rothamsted-electronic-archive" class="section level3">
<h3>Rothamsted Electronic Archive</h3>
<p>http://www.era.rothamsted.ac.uk/index.php Data from Broadbalk and
other long-term experiments.</p>
<p>Github draft data:
https://github.com/Rothamsted-Ecoinformatics/YieldbookDatasetDrafts</p>
</div>
<div id="rothamsted-documents-archive" class="section level3">
<h3>Rothamsted Documents Archive</h3>
<p>http://www.era.rothamsted.ac.uk/eradoc/collections.php</p>
<p>Annual reports from Rothamsted 1908-1987. Many have data, especially
in the early years (before WWII) there are data given for the ‘Classical
Experiments’.</p>
<pre><code>Year, page
1908-1926 
1926-1927 agridat::sawyer.multi.uniformity 
1927-1928 agridat::sawyer.multi.uniformity
1929-1930
1931,143 agridat::yates.oats
1932
1933
1934,215-222 Sugar beet multi-environment trial with 3^3 fertilizer 
  treatments at each site Roots, SugarPercent, SugarWeight, PlantNumber, Tops, Purity.
1935
1936,241 Similar to the 1934 experiment, but only gives the main effects, 
  not the actual data.
1937-1939
1946-1955

1986</code></pre>
</div>
<div id="yates-1937-the-design-and-analysis-of-factorial-experiments" class="section level3">
<h3>Yates (1937), The Design and analysis of factorial experiments</h3>
<pre><code>9 2x2x2, 4 rep
27 2x2x2x2x2 factorial
33 2x2x2 factorial in two 4x4 Latin Squares
42 3x3x3 factorial
59 3x2x2 factorial in 3 reps.  See also page 39.
74 Split-plot agridat::yates.oats</code></pre>
</div>
<div id="statistical-analysis-of-agricultural-experiments-with-r" class="section level3">
<h3>Statistical Analysis of Agricultural Experiments with R</h3>
<p>rstats4ag.org (no http included here because of firewall
problems).</p>
<p>Datasets for mixed models, ancova, dose response curves,
competition.</p>
</div>
<div id="syngenta-crop-challenge" class="section level3">
<h3>Syngenta Crop Challenge</h3>
<p>https://www.ideaconnection.com/syngenta-crop-challenge/</p>
<p>Annual Kaggle-style competition sponsored by Syngenta.</p>
</div>
<div id="terra-ref" class="section level3">
<h3>Terra-Ref</h3>
<p>https://terraref.org/</p>
<p>Sensor observations, plant phenotypes, derived traits, genetic and
genomic data. Beta version until Nov 2018.</p>
</div>
<div id="usda-national-agricultural-statistics-service" class="section level3">
<h3>USDA National Agricultural Statistics Service</h3>
<p>https://www.nass.usda.gov https://quickstats.nass.usda.gov/</p>
<p>Group: Field Crops Commodity: Corn Category: Area Harvested, Yield
Data Item: Corn grain Acres Harvested, Yield Bu/Ac Domain: Total
Geography: State See <code>agridat::nass.corn</code>, nass.wheat,
etc.</p>
</div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
